diff --git a/CMakeLists.txt b/CMakeLists.txt
index 12bcf1031..9f776f8d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,23 +19,16 @@
 # Project
 ####################################################
 
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.22)
 
-if(${CMAKE_VERSION} VERSION_LESS "3.22")
-    message(STATUS "This project recommends using CMake version 3.22 or higher. You are using ${CMAKE_VERSION}.")
-else()
-    cmake_policy(SET CMP0128 NEW)
-endif()
-
-if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
-    message(FATAL_ERROR "In-source builds are not allowed.\
-     Run cmake from a separate directory from where CMakeLists.txt lives.\
-     NOTE: cmake will now create CMakeCache.txt and CMakeFiles/*.\
-     You must delete them, or cmake will refuse to work.")
-endif()
+# https://cmake.org/cmake/help/latest/policy/CMP0128.html
+# if this is not set, CMake will use OLD behavior
+cmake_policy(SET CMP0128 NEW)
 
 project(wolfssl VERSION 5.8.4 LANGUAGES C ASM)
 
+set(CMAKE_DEBUG_POSTFIX "d" CACHE STRING "Filename postfix for libraries under DEBUG configuration")
+
 # Set WOLFSSL_ROOT if not already defined
 if ("${WOLFSSL_ROOT}" STREQUAL "")
     # we'll assume this CMakeLists.txt is in the root of wolfSSL
@@ -66,9 +59,6 @@ set(WOLFSSL_DEFINITIONS)
 set(WOLFSSL_LINK_LIBS)
 set(WOLFSSL_INCLUDE_DIRS)
 
-# Initialize pkg-config private variables
-set(PC_LIBS_PRIVATE "")
-
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/")
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/functions.cmake)
 
@@ -251,8 +241,6 @@ if(WOLFSSL_REPRODUCIBLE_BUILD)
     set(CMAKE_C_ARCHIVE_FINISH "<CMAKE_RANLIB> -D <TARGET>")
 endif()
 
-add_option("WOLFSSL_INSTALL" "Create install target for WolfSSL project" "yes" "yes;no")
-
 # Support for forcing 32-bit mode
 # TODO: detect platform from other options
 add_option("WOLFSSL_32BIT"
@@ -2428,21 +2416,15 @@ file(REMOVE ${OPTION_FILE})
 
 # TODO: - Build shared/static libs based on enables. Check CMake
 #         global flag BUILD_SHARED_LIBS.
-option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static ones (.lib/.a)" ON)
+option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static ones (.lib/.a)" NO)
 
 set(LIB_SOURCES "")
 # Generates a list of sources to include in the library.
 # Corresponds to the instances of "src_libwolfssl_la_SOURCES += ..."
 # in the *.am files.
 generate_lib_src_list("${LIB_SOURCES}")
-if(BUILD_SHARED_LIBS)
-    message(STATUS "BUILD_SHARED_LIBS enabled: ${LIB_SOURCES}")
-    add_library(wolfssl SHARED ${LIB_SOURCES})
-else()
-    message(STATUS "Static Libs: ${LIB_SOURCES}")
-    add_library(wolfssl STATIC ${LIB_SOURCES})
-endif()
 
+add_library(wolfssl ${LIB_SOURCES})
 add_library(wolfssl::wolfssl ALIAS wolfssl)
 
 if (NOT "$ENV{ARIA_DIR}" STREQUAL "")
@@ -2857,7 +2839,6 @@ else()
   endif()
 endif()
 
-
 if(NOT BUILD_INTEL_QA AND NOT BUILD_INTEL_QA_SYNC)
   list(APPEND HEADER_EXCLUDE
     "wolfssl/wolfcrypt/port/intel")
@@ -2906,109 +2887,31 @@ list(JOIN HEADER_EXCLUDE "|" EXCLUDED_HEADERS_REGEX)
 string(PREPEND EXCLUDED_HEADERS_REGEX "(")
 string(APPEND  EXCLUDED_HEADERS_REGEX ")")
 
-if(WOLFSSL_INSTALL)
-
-    set(INSTALLED_EXAMPLES
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/echoserver/echoserver.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/sctp/sctp-server.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/sctp/sctp-client-dtls.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/sctp/sctp-client.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/sctp/sctp-server-dtls.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/echoclient/echoclient.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/server/server.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/benchmark/tls_bench.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/examples/client/client.c)
-
-
-    # Install the library
-    install(TARGETS wolfssl
-            EXPORT wolfssl-targets
-            LIBRARY DESTINATION lib
-            ARCHIVE DESTINATION lib
-            RUNTIME DESTINATION bin
-            )
-    # Install the headers
-    install(DIRECTORY ${WOLFSSL_OUTPUT_BASE}/wolfssl/
-            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/wolfssl
-            FILES_MATCHING PATTERN "*.h"
-            REGEX ${EXCLUDED_HEADERS_REGEX} EXCLUDE)
-    install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/
-            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/wolfssl
-            FILES_MATCHING PATTERN "*.h"
-            REGEX ${EXCLUDED_HEADERS_REGEX} EXCLUDE)
-
-    # Install the examples
-    install(FILES ${INSTALLED_EXAMPLES}
-            DESTINATION ${CMAKE_INSTALL_DOCDIR}/example)
-    # Install README.txt and taoCert.txt
-    install(FILES
-            ${CMAKE_CURRENT_SOURCE_DIR}/doc/README.txt
-            ${CMAKE_CURRENT_SOURCE_DIR}/certs/taoCert.txt
-            DESTINATION ${CMAKE_INSTALL_DOCDIR})
-    # Install the export set
-    install(EXPORT wolfssl-targets
-            DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/wolfssl
-            FILE wolfssl-targets.cmake
-            NAMESPACE wolfssl::)
-
-    # TODO: Distro build + rules for what to include in the distro.
-    #       See various include.am files.
-
-    set(prefix ${CMAKE_INSTALL_PREFIX})
-    set(exec_prefix "\${prefix}")
-    set(libdir "\${exec_prefix}/lib")
-    set(includedir "\${prefix}/include")
-    set(VERSION ${PROJECT_VERSION})
-
-    if(CMAKE_C_COMPILER_ID STREQUAL "OpenWatcom")
-    else()
-        # Setting libm in Libs.private of wolfssl.pc.
-        # See "Link Libraries" in above about `m` insertion to LINK_LIBRARIES
-        get_target_property(_wolfssl_dep_libs wolfssl LINK_LIBRARIES)
-        list(FIND _wolfssl_dep_libs m _dep_libm)
-        if ("${_dep_libm}" GREATER -1)
-            set(LIBM -lm)
-        else()
-            set(LIBM)
-        endif()
-    endif()
-
-    # Add required frameworks for static linking on Apple platforms
-    if(APPLE AND NOT BUILD_SHARED_LIBS)
-        if(WOLFSSL_SYS_CA_CERTS)
-            list(APPEND PC_LIBS_PRIVATE "-framework CoreFoundation" "-framework Security")
-        endif()
-    endif()
-
-    # Convert lists to space-separated strings for pkg-config
-    string(JOIN " " PC_LIBS_PRIVATE ${PC_LIBS_PRIVATE})
-
-    configure_file(support/wolfssl.pc.in ${CMAKE_CURRENT_BINARY_DIR}/support/wolfssl.pc @ONLY)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/support/wolfssl.pc
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
-
-    include(CMakePackageConfigHelpers)
-    configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Config.cmake.in
-    "${CMAKE_CURRENT_BINARY_DIR}/wolfssl-config.cmake"
-    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/wolfssl"
-    NO_SET_AND_CHECK_MACRO
-    NO_CHECK_REQUIRED_COMPONENTS_MACRO
-    )
-
-    export(EXPORT wolfssl-targets
-    FILE "${CMAKE_CURRENT_BINARY_DIR}/wolfssl-targets.cmake"
-    NAMESPACE wolfssl::
-    )
-
-    write_basic_package_version_file(
-    "${CMAKE_CURRENT_BINARY_DIR}/wolfssl-config-version.cmake"
-    VERSION "${wolfssl_VERSION_MAJOR}.${wolfssl_VERSION_MINOR}"
-    COMPATIBILITY AnyNewerVersion
-    )
+set(PROJECT_NAMESPACE_FOR_PACKAGE "wolfssl")
+include("${CMAKE_CURRENT_SOURCE_DIR}/Installing.cmake")
+
+# headers
+install(
+    DIRECTORY
+        ${WOLFSSL_OUTPUT_BASE}/wolfssl/
+    DESTINATION
+        ${CMAKE_INSTALL_INCLUDEDIR}/wolfssl
+    FILES_MATCHING PATTERN "*.h"
+    REGEX ${EXCLUDED_HEADERS_REGEX} EXCLUDE
+)
+install(
+    DIRECTORY
+        ${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/
+    DESTINATION
+        ${CMAKE_INSTALL_INCLUDEDIR}/wolfssl
+    FILES_MATCHING PATTERN "*.h"
+    REGEX ${EXCLUDED_HEADERS_REGEX} EXCLUDE
+)
 
-    install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/wolfssl-config.cmake
-    ${CMAKE_CURRENT_BINARY_DIR}/wolfssl-config-version.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/wolfssl
-    )
-endif()
+# documentation
+install(
+    FILES
+        ${CMAKE_CURRENT_SOURCE_DIR}/certs/taoCert.txt
+    DESTINATION
+        "share/${PROJECT_NAME}/doc"
+)
diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
deleted file mode 100644
index 3b8098b6b..000000000
--- a/cmake/Config.cmake.in
+++ /dev/null
@@ -1,9 +0,0 @@
-@PACKAGE_INIT@
-
-include(CMakeFindDependencyMacro)
-if (@HAVE_PTHREAD@)
-  find_dependency(Threads)
-endif()
-
-
-include ( "${CMAKE_CURRENT_LIST_DIR}/wolfssl-targets.cmake" )
diff --git a/zephyr/CMakeLists.txt b/zephyr/CMakeLists.txt
index aeb714b94..b2fae0c23 100644
--- a/zephyr/CMakeLists.txt
+++ b/zephyr/CMakeLists.txt
@@ -197,12 +197,12 @@ if(CONFIG_WOLFSSL)
     # therefore susceptible to bit rot
 
     target_include_directories(wolfSSL INTERFACE
-        ${CONFIG_WOLFSSL_INSTALL_PATH}
+        ${CONFIG_WOLFSSL_INSTALL_PATH} # where is it defined?
         )
 
     zephyr_link_libraries(
       wolfssl_external
-      -L${CONFIG_WOLFSSL_INSTALL_PATH}
+      -L${CONFIG_WOLFSSL_INSTALL_PATH} # where is it defined?
       gcc
       )
     # Lib wolfssl depends on libgcc so to allow
