diff --git a/wolfcrypt/src/asn.c b/wolfcrypt/src/asn.c
index e54c9cc..6dcc543 100644
--- a/wolfcrypt/src/asn.c
+++ b/wolfcrypt/src/asn.c
@@ -27592,6 +27592,6 @@ static int KeyPemToDerPassCb(char* passwd, int sz, int rw, void* userdata)
         return 0;
 
-    XSTRNCPY(passwd, (char*)userdata, (size_t)sz);
-    return (int)min((word32)sz, (word32)XSTRLEN((char*)userdata));
+    XSTRLCPY(passwd, (char*)userdata, (size_t)sz);
+    return (int)min((word32)(sz - 1), (word32)XSTRLEN((char*)userdata));
 }
 #endif
