diff --git a/src/microsoft/spirv_to_dxil/meson.build b/src/microsoft/spirv_to_dxil/meson.build
index 7a75e267760..eea9d146286 100644
--- a/src/microsoft/spirv_to_dxil/meson.build
+++ b/src/microsoft/spirv_to_dxil/meson.build
@@ -19,28 +19,53 @@
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 # IN THE SOFTWARE.
 
-libspirv_to_dxil = both_libraries(
-  'spirv_to_dxil',
-  [
-    files(
-      'dxil_spirv_nir.c',
-      'dxil_spirv_nir.h',
-      'dxil_spirv_nir_lower_bindless.c',
-      'spirv_to_dxil.c',
-      'spirv_to_dxil.h',
-    ),
-    sha1_h,
-  ],
-  vs_module_defs : 'spirv_to_dxil.def',
-  dependencies : [idep_nir, idep_vtn, idep_libdxil_compiler],
-  include_directories : [inc_include, inc_src],
-  install : true,
-)
+if get_option('default_library') == 'static'
+  libspirv_to_dxil = library(
+    'spirv_to_dxil',
+    [
+      files(
+        'dxil_spirv_nir.c',
+        'dxil_spirv_nir.h',
+        'dxil_spirv_nir_lower_bindless.c',
+        'spirv_to_dxil.c',
+        'spirv_to_dxil.h',
+      ),
+      sha1_h,
+    ],
+    vs_module_defs : 'spirv_to_dxil.def',
+    dependencies : [idep_nir, idep_vtn, idep_libdxil_compiler],
+    include_directories : [inc_include, inc_src],
+    install : true,
+  )
 
-idep_libspirv_to_dxil = declare_dependency(
-   link_with : libspirv_to_dxil.get_static_lib(),
-   include_directories : include_directories('.')
-)
+  idep_libspirv_to_dxil = declare_dependency(
+     link_with : libspirv_to_dxil,
+     include_directories : include_directories('.')
+  )
+else
+  libspirv_to_dxil = both_libraries(
+    'spirv_to_dxil',
+    [
+      files(
+        'dxil_spirv_nir.c',
+        'dxil_spirv_nir.h',
+        'dxil_spirv_nir_lower_bindless.c',
+        'spirv_to_dxil.c',
+        'spirv_to_dxil.h',
+      ),
+      sha1_h,
+    ],
+    vs_module_defs : 'spirv_to_dxil.def',
+    dependencies : [idep_nir, idep_vtn, idep_libdxil_compiler],
+    include_directories : [inc_include, inc_src],
+    install : true,
+  )
+
+  idep_libspirv_to_dxil = declare_dependency(
+     link_with : libspirv_to_dxil.get_static_lib(),
+     include_directories : include_directories('.')
+  )
+endif
 
 spirv_to_dxil = executable(
   'spirv2dxil',
