diff --git a/CMakeLists.txt b/CMakeLists.txt
index 26dc8e74..d87c0612 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,14 +16,16 @@
 
 # https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md
 # As of 2024-07-01, CMake 3.16 is the minimum supported version.
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.28)
 
 # Allow the user to specify the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT
 if (POLICY CMP0141)
   cmake_policy(SET CMP0141 NEW)
 endif (POLICY CMP0141)
 
-project(absl LANGUAGES CXX VERSION 20260107)
+set(ABSEIL_VERSION "0.0.0" CACHE STRING "Abseil version")
+
+project(absl LANGUAGES CXX VERSION "${ABSEIL_VERSION}")
 set(ABSL_SOVERSION "2601.0.0")
 include(CTest)
 
@@ -41,6 +43,8 @@ else()
   option(ABSL_ENABLE_INSTALL "Enable install rule" ON)
 endif()
 
+set(CMAKE_DEBUG_POSTFIX "d" CACHE STRING "Filename postfix for libraries under DEBUG configuration")
+
 set(CMAKE_INSTALL_RPATH "$ORIGIN")
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 set(CMAKE_BUILD_RPATH_USE_ORIGIN ON)
