diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d3059d5..f10437b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,14 +16,16 @@
 
 # https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md
 # As of 2024-07-01, CMake 3.16 is the minimum supported version.
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.22)
 
 # Allow the user to specify the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT
 if (POLICY CMP0141)
   cmake_policy(SET CMP0141 NEW)
 endif (POLICY CMP0141)
 
-project(absl LANGUAGES CXX VERSION 20250512)
+set(ABSEIL_VERSION "0.0.0" CACHE STRING "Abseil version")
+
+project(absl LANGUAGES CXX VERSION "${ABSEIL_VERSION}")
 set(ABSL_SOVERSION "2505.0.0")
 include(CTest)
 
