diff --git a/CMakeLists.txt b/CMakeLists.txt
index d88feb5c..61b98642 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -202,8 +202,8 @@ if(JSON_Install)
         NAMESPACE ${PROJECT_NAME}::
         DESTINATION ${NLOHMANN_JSON_CONFIG_INSTALL_DIR}
     )
-    install(
-        FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
-        DESTINATION ${NLOHMANN_JSON_PKGCONFIG_INSTALL_DIR}
-    )
+    #install(
+    #    FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
+    #    DESTINATION ${NLOHMANN_JSON_PKGCONFIG_INSTALL_DIR}
+    #)
 endif()
