diff --git a/CMakeLists.txt b/CMakeLists.txt
index 150b0129..f6817158 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -365,8 +365,8 @@ if (GLSLANG_ENABLE_INSTALL)
         include("@PACKAGE_PATH_EXPORT_TARGETS@")
     ]=])
 
-    set(PATH_EXPORT_TARGETS "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}/glslang-targets.cmake")
-    if(UNIX OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia")
+    set(PATH_EXPORT_TARGETS "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}/glslang-targets.cmake")
+    if((UNIX OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia") AND NOT APPLE) # NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin"
         set(INSTALL_CONFIG_UNIX [=[
             set(THREADS_PREFER_PTHREAD_FLAG ON)
             find_dependency(Threads)
@@ -377,7 +377,7 @@ if (GLSLANG_ENABLE_INSTALL)
         "${CMAKE_CURRENT_BINARY_DIR}/glslang-config.cmake"
         PATH_VARS
             PATH_EXPORT_TARGETS
-        INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+        INSTALL_DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}
     )
 
     write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/glslang-config-version.cmake"
@@ -388,7 +388,7 @@ if (GLSLANG_ENABLE_INSTALL)
     install(
         EXPORT      glslang-targets
         NAMESPACE   "glslang::"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+        DESTINATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}"
     )
 
     install(
@@ -396,6 +396,6 @@ if (GLSLANG_ENABLE_INSTALL)
             "${CMAKE_CURRENT_BINARY_DIR}/glslang-config.cmake"
             "${CMAKE_CURRENT_BINARY_DIR}/glslang-config-version.cmake"
         DESTINATION
-            "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+            "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}"
     )
 endif()
diff --git a/glslang/CMakeLists.txt b/glslang/CMakeLists.txt
index eb1dc68e..9d905a08 100644
--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -197,7 +197,7 @@ if(ENABLE_SPIRV)
 endif()
 
 # Link pthread
-if(UNIX OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia" OR ANDROID)
+if((UNIX OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Fuchsia" OR ANDROID) AND NOT APPLE) # NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin"
     set(THREADS_PREFER_PTHREAD_FLAG ON)
     find_package(Threads REQUIRED)
     target_link_libraries(glslang Threads::Threads)
@@ -260,11 +260,9 @@ if(GLSLANG_ENABLE_INSTALL)
         Public/ResourceLimits.h
         Public/ShaderLang.h
         Public/resource_limits_c.h
-        Include/glslang_c_interface.h
-        Include/glslang_c_shader_types.h
-        Include/ResourceLimits.h
-        Include/visibility.h
-        MachineIndependent/Versions.h)
+        ${GLSLANG_HEADERS}
+        ${MACHINEINDEPENDENT_HEADERS}
+    )
 
     foreach(file ${PUBLIC_HEADERS})
         get_filename_component(dir ${file} DIRECTORY)
