diff --git a/glslang/OSDependent/Unix/CMakeLists.txt b/glslang/OSDependent/Unix/CMakeLists.txt
index f6b1c6af..9a10d78b 100644
--- a/glslang/OSDependent/Unix/CMakeLists.txt
+++ b/glslang/OSDependent/Unix/CMakeLists.txt
@@ -37,8 +37,10 @@ set_property(TARGET OSDependent PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 # Link pthread
 set(THREADS_PREFER_PTHREAD_FLAG ON)
+if(NOT APPLE)
 find_package(Threads REQUIRED)
 target_link_libraries(OSDependent Threads::Threads)
+endif()
 
 if(ENABLE_GLSLANG_INSTALL AND NOT BUILD_SHARED_LIBS)
     install(TARGETS OSDependent EXPORT glslang-targets)
