diff --git a/CMakeLists.txt b/CMakeLists.txt
index 438055d..5b3c42a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,8 @@ draco_set_cxx_flags()
 draco_set_exe_linker_flags()
 draco_generate_features_h()
 
+option(DRACO_WITH_TOOLS "Build tools" 0)
+
 # Draco source file listing variables.
 list(APPEND draco_attributes_sources
             "${draco_src_root}/attributes/attribute_octahedron_transform.cc"
@@ -663,6 +665,7 @@ if(EMSCRIPTEN AND DRACO_JS_GLUE)
                 "DRACO_ATTRIBUTE_VALUES_DEDUPLICATION_SUPPORTED")
   endif()
 
+  if(DRACO_WITH_TOOLS)
   draco_add_emscripten_executable(NAME
                                   draco_decoder
                                   SOURCES
@@ -754,6 +757,7 @@ if(EMSCRIPTEN AND DRACO_JS_GLUE)
       POST_LINK_JS_SOURCES
       ${draco_post_link_js_sources})
   endif()
+  endif()
 else()
   # Standard Draco libs, encoder and decoder. Object collections that mirror the
   # Draco directory structure.
@@ -1063,6 +1067,7 @@ else()
     endif()
   endif()
 
+  if(DRACO_WITH_TOOLS)
   # Draco app targets.
   draco_add_executable(NAME
                        draco_decoder
@@ -1103,6 +1108,7 @@ else()
                          LIB_DEPS
                          ${draco_dependency})
   endif()
+  endif()
 
   draco_setup_install_target()
   draco_setup_test_targets()
diff --git a/cmake/draco_emscripten.cmake b/cmake/draco_emscripten.cmake
index 08c58d6..a24444e 100644
--- a/cmake/draco_emscripten.cmake
+++ b/cmake/draco_emscripten.cmake
@@ -192,6 +192,7 @@ macro(draco_add_emscripten_executable)
   # passed in with the target.
   list(APPEND emexe_LINK_FLAGS ${DRACO_CXX_FLAGS})
 
+  if(DRACO_WITH_TOOLS)
   if(DRACO_GLTF_BITSTREAM)
     # Add "_gltf" suffix to target output name.
     draco_add_executable(NAME
@@ -211,6 +212,7 @@ macro(draco_add_emscripten_executable)
                          ${emexe_DEFINES} INCLUDES ${emexe_INCLUDES} LINK_FLAGS
                          ${emexe_LINK_FLAGS})
   endif()
+  endif()
 
   foreach(feature ${emexe_FEATURES})
     draco_enable_feature(FEATURE ${feature} TARGETS ${emexe_NAME})
diff --git a/cmake/draco_install.cmake b/cmake/draco_install.cmake
index e15a054..f4d996f 100644
--- a/cmake/draco_install.cmake
+++ b/cmake/draco_install.cmake
@@ -51,10 +51,12 @@ macro(draco_setup_install_target)
     FILES "${draco_build}/draco/draco_features.h"
     DESTINATION "${includes_path}/draco/")
 
+  if(DRACO_WITH_TOOLS)
   install(TARGETS draco_decoder DESTINATION "${bin_path}")
   install(TARGETS draco_encoder DESTINATION "${bin_path}")
+  endif()
 
-  if(DRACO_TRANSCODER_SUPPORTED)
+  if(DRACO_TRANSCODER_SUPPORTED AND DRACO_WITH_TOOLS)
     install(TARGETS draco_transcoder DESTINATION "${bin_path}")
   endif()
 
@@ -90,9 +92,9 @@ macro(draco_setup_install_target)
   endif()
 
   # pkg-config: draco.pc
-  configure_file("${draco_root}/cmake/draco.pc.template"
-                 "${draco_build}/draco.pc" @ONLY NEWLINE_STYLE UNIX)
-  install(FILES "${draco_build}/draco.pc" DESTINATION "${libs_path}/pkgconfig")
+  # configure_file("${draco_root}/cmake/draco.pc.template"
+  #                "${draco_build}/draco.pc" @ONLY NEWLINE_STYLE UNIX)
+  # install(FILES "${draco_build}/draco.pc" DESTINATION "${libs_path}/pkgconfig")
 
   # CMake config: draco-config.cmake
   configure_package_config_file(
