diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 676ee979..875cc9ba 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -46,8 +46,14 @@ if (IS_DIRECTORY ${SPIRV_HEADER_DIR})
     add_subdirectory(${SPIRV_HEADER_DIR})
   endif()
 else()
-  message(FATAL_ERROR
-    "SPIRV-Headers was not found - please checkout a copy under external/.")
+  find_package(SPIRV-Headers CONFIG REQUIRED)
+  get_target_property(
+    SPIRV_HEADERS_INCLUDE_PATH
+    "SPIRV-Headers::SPIRV-Headers"
+    INTERFACE_INCLUDE_DIRECTORIES
+  )
+  set(SPIRV_HEADER_INCLUDE_DIR ${SPIRV_HEADERS_INCLUDE_PATH} PARENT_SCOPE)
+  #target_link_libraries(${PROJECT_NAME} PRIVATE SPIRV-Headers::SPIRV-Headers)
 endif()
 
 if (NOT ${SPIRV_SKIP_TESTS})
