diff --git a/implot.h b/implot.h
index 36c0d40..f30c95f 100644
--- a/implot.h
+++ b/implot.h
@@ -45,7 +45,7 @@
 // [SECTION] Obsolete API
 
 #pragma once
-#include "imgui.h"
+#include <dearimgui/imgui.h>
 
 //-----------------------------------------------------------------------------
 // [SECTION] Macros and Defines
diff --git a/implot_internal.h b/implot_internal.h
index fb01204..e7b6606 100644
--- a/implot_internal.h
+++ b/implot_internal.h
@@ -32,7 +32,7 @@
 #pragma once
 
 #include <time.h>
-#include "imgui_internal.h"
+#include <dearimgui/imgui_internal.h>
 
 #ifndef IMPLOT_VERSION
 #error Must include implot.h before implot_internal.h
