diff --git a/implot.h b/implot.h
index 2ab2f40..bbe6163 100644
--- a/implot.h
+++ b/implot.h
@@ -46,7 +46,7 @@
 // [SECTION] Obsolete API
 
 #pragma once
-#include "imgui.h"
+#include <dearimgui/imgui.h>
 #ifndef IMGUI_DISABLE
 
 //-----------------------------------------------------------------------------
diff --git a/implot_internal.h b/implot_internal.h
index 46dfaa5..3d8e7aa 100644
--- a/implot_internal.h
+++ b/implot_internal.h
@@ -38,7 +38,7 @@
 
 #ifndef IMGUI_DISABLE
 #include <time.h>
-#include "imgui_internal.h"
+#include <dearimgui/imgui_internal.h>
 
 // Support for pre-1.84 versions. ImPool's GetSize() -> GetBufSize()
 #if (IMGUI_VERSION_NUM < 18303)
