diff --git a/googletest/cmake/internal_utils.cmake b/googletest/cmake/internal_utils.cmake
index 41405587..d3196fd7 100644
--- a/googletest/cmake/internal_utils.cmake
+++ b/googletest/cmake/internal_utils.cmake
@@ -345,12 +345,12 @@ function(install_project)
       endforeach()
     endif()
     # Configure and install pkgconfig files.
-    foreach(t ${ARGN})
-      set(configured_pc "${generated_dir}/${t}.pc")
-      configure_file("${PROJECT_SOURCE_DIR}/cmake/${t}.pc.in"
-        "${configured_pc}" @ONLY)
-      install(FILES "${configured_pc}"
-        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    endforeach()
+    #foreach(t ${ARGN})
+    #  set(configured_pc "${generated_dir}/${t}.pc")
+    #  configure_file("${PROJECT_SOURCE_DIR}/cmake/${t}.pc.in"
+    #    "${configured_pc}" @ONLY)
+    #  install(FILES "${configured_pc}"
+    #    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+    #endforeach()
   endif()
 endfunction()
