diff --git a/CMakeLists.txt b/CMakeLists.txt
index c35d2eb3..c92d870d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -180,10 +180,10 @@ if(SIMDJSON_ENABLE_THREADS)
 endif()
 
 configure_file("simdjson.pc.in" "simdjson.pc" @ONLY)
-install(
-    FILES "${CMAKE_CURRENT_BINARY_DIR}/simdjson.pc"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
-)
+# install(
+#     FILES "${CMAKE_CURRENT_BINARY_DIR}/simdjson.pc"
+#     DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+# )
 
 #
 # CPack
