diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca23a8a0..98ecf71e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -163,7 +163,7 @@ include(GNUInstallDirs)
 if(SIMDJSON_SINGLEHEADER)
   install(
       FILES singleheader/simdjson.h
-      DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+      DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}"
       COMPONENT simdjson_Development
   )
 endif()
