diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0e1cfe6c..b47c01c1f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ set(PNGLIB_VERSION ${PNGLIB_MAJOR}.${PNGLIB_MINOR}.${PNGLIB_REVISION})
 set(PNGLIB_ABI_VERSION ${PNGLIB_MAJOR}${PNGLIB_MINOR})
 set(PNGLIB_SHARED_VERSION ${PNGLIB_ABI_VERSION}.${PNGLIB_REVISION}.${PNGLIB_SUBREVISION})
 
-project(libpng
+project(png
         VERSION ${PNGLIB_VERSION}
         LANGUAGES C)
 
@@ -70,30 +70,23 @@ if((NOT DFA_XTRA STREQUAL "") AND (NOT PNG_LIBCONF_HEADER STREQUAL ""))
 endif()
 
 # Allow the users to switch on/off various library build types.
-option(PNG_SHARED "Build libpng as a shared library" ON)
-option(PNG_STATIC "Build libpng as a static library" ON)
-if(APPLE)
-  option(PNG_FRAMEWORK "Build libpng as a framework bundle" ON)
-else()
-  option(PNG_FRAMEWORK "Build libpng as a framework bundle (not available on this platform)" OFF)
-endif()
-if(NOT APPLE AND PNG_FRAMEWORK)
-  message(SEND_ERROR "The option PNG_FRAMEWORK should not be set on this platform")
+option(PNG_STATIC "Build ${PROJECT_NAME} as a static library" 1)
+option(${PROJECT_NAME} "Build ${PROJECT_NAME} as a shared library" 0)
+option(PNG_FRAMEWORK "Build libpng as a framework bundle" 0)
+
+if(PNG_STATIC AND PNG_SHARED)
+    message(FATAL_ERROR "Cannot build STATIC and SHARED variant at the same time, you need to choose one")
 endif()
 
 # Allow the users to switch on/off the auxiliary build and test artifacts.
 # These artifacts are NOT part of libpng proper, and are subject to change
 # at any time.
-option(PNG_TESTS "Build the libpng tests" ON)
+option(PNG_TESTS "Build the libpng tests" 0)
 
 # Same as above, but for the third-party tools.
 # Although these tools are targetted at development environments only,
 # the users are allowed to override the option to build by default.
-if(ANDROID OR IOS)
-  option(PNG_TOOLS "Build the libpng tools" OFF)
-else()
-  option(PNG_TOOLS "Build the libpng tools" ON)
-endif()
+option(PNG_TOOLS "Build the libpng tools" 0)
 
 # Maintain backwards compatibility with the deprecated option PNG_EXECUTABLES.
 option(PNG_EXECUTABLES "[Deprecated; please use PNG_TOOLS]" ON)
@@ -124,22 +117,10 @@ else()
 endif()
 message(STATUS "Building for target architecture: ${PNG_TARGET_ARCHITECTURE}")
 
-# Allow the users to specify a custom location of zlib.
-# With CMake 3.12 and newer, this option is no longer necessary.
-option(PNG_BUILD_ZLIB "[Deprecated; please use ZLIB_ROOT]" OFF)
-if(PNG_BUILD_ZLIB)
-  if("x${ZLIB_ROOT}" STREQUAL "x")
-    message(SEND_ERROR "The option PNG_BUILD_ZLIB=${PNG_BUILD_ZLIB} is no longer supported; "
-                       "please use ZLIB_ROOT instead")
-  else()
-    message(SEND_ERROR "The option PNG_BUILD_ZLIB=${PNG_BUILD_ZLIB} is no longer supported; "
-                       "using ZLIB_ROOT=\"${ZLIB_ROOT}\"")
-  endif()
-endif()
-
-# Find the zlib library.
-find_package(ZLIB REQUIRED)
-set(PNG_LINK_LIBRARIES ZLIB::ZLIB)
+find_package(zlib CONFIG REQUIRED)
+message(STATUS "zlib version: ${zlib_VERSION}")
+# a crutch for their crazy version definitions mechanism
+set(ZLIB_INCLUDE_DIRS "${zlib_DIR}/../../include")
 
 # Find the math library (unless we already know it's not available or
 # not needed).
@@ -642,111 +623,89 @@ set(png_fix_itxt_sources
 # Initialize the list of libpng library targets.
 set(PNG_LIBRARY_TARGETS "")
 
-# Initialize the libpng library file names.
-if(UNIX
-   OR (WIN32 AND NOT CMAKE_SHARED_LIBRARY_PREFIX STREQUAL "")
-   OR (WIN32 AND NOT CMAKE_STATIC_LIBRARY_PREFIX STREQUAL ""))
-  # We are on a Unix or Unix-like toolchain like the GNU toolchain on Windows.
-  # Library file names are expected to have an implicit prefix such as "lib".
-  # Let CMake prepend and append its usual prefixes and suffixes by default.
-  set(PNG_SHARED_OUTPUT_NAME "png${PNGLIB_ABI_VERSION}")
-  set(PNG_STATIC_OUTPUT_NAME "png${PNGLIB_ABI_VERSION}")
-else()
-  # We are, most likely, on a Windows toolchain like MSVC, Clang on Windows,
-  # Borland/Embarcadero, etc. We need to specify the "libpng" name explicitly.
-  # We also need to use a custom suffix, in order to distinguish between the
-  # shared import library name and the static library name.
-  set(PNG_SHARED_OUTPUT_NAME "libpng${PNGLIB_ABI_VERSION}")
-  set(PNG_STATIC_OUTPUT_NAME "libpng${PNGLIB_ABI_VERSION}_static")
-endif()
+set(PNG_SHARED_OUTPUT_NAME "${PROJECT_NAME}")
 
 if(PNG_SHARED)
-  add_library(png_shared SHARED ${libpng_sources})
-  add_dependencies(png_shared png_genfiles)
-  list(APPEND PNG_LIBRARY_TARGETS png_shared)
-  set_target_properties(png_shared
+    add_library(${PROJECT_NAME} SHARED ${libpng_sources})
+    add_dependencies(${PROJECT_NAME} png_genfiles)
+    list(APPEND PNG_LIBRARY_TARGETS ${PROJECT_NAME})
+    set_target_properties(${PROJECT_NAME}
                         PROPERTIES OUTPUT_NAME "${PNG_SHARED_OUTPUT_NAME}"
                                    DEBUG_POSTFIX "${PNG_DEBUG_POSTFIX}"
                                    VERSION "${PNGLIB_SHARED_VERSION}"
                                    SOVERSION "${PNGLIB_ABI_VERSION}")
   if(UNIX AND AWK)
     if(HAVE_LD_VERSION_SCRIPT)
-      set_target_properties(png_shared
+      set_target_properties(${PROJECT_NAME}
                             PROPERTIES LINK_FLAGS "-Wl,--version-script='${CMAKE_CURRENT_BINARY_DIR}/libpng.vers'")
     elseif(HAVE_SOLARIS_LD_VERSION_SCRIPT)
-      set_target_properties(png_shared
+      set_target_properties(${PROJECT_NAME}
                             PROPERTIES LINK_FLAGS "-Wl,-M -Wl,'${CMAKE_CURRENT_BINARY_DIR}/libpng.vers'")
     endif()
   endif()
   if(APPLE)
     # Avoid CMake's implicit compile definition "png_shared_EXPORTS".
-    set_target_properties(png_shared
+    set_target_properties(${PROJECT_NAME}
                           PROPERTIES DEFINE_SYMBOL "")
   elseif(WIN32)
     # Use the explicit compile definition "PNG_BUILD_DLL" for Windows DLLs.
-    set_target_properties(png_shared
+    set_target_properties(${PROJECT_NAME}
                           PROPERTIES DEFINE_SYMBOL PNG_BUILD_DLL)
   endif()
-  target_include_directories(png_shared
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
-  target_include_directories(png_shared
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
-  target_include_directories(png_shared
+  target_include_directories(${PROJECT_NAME}
                              SYSTEM
-                             INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/libpng${PNGLIB_ABI_VERSION}>")
-  target_link_libraries(png_shared
-                        PUBLIC ${PNG_LINK_LIBRARIES})
+                             INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+  target_link_libraries(${PROJECT_NAME}
+                        PUBLIC zlib ${M_LIBRARY})
 endif()
 
 if(PNG_STATIC)
-  add_library(png_static STATIC ${libpng_sources})
-  add_dependencies(png_static png_genfiles)
-  list(APPEND PNG_LIBRARY_TARGETS png_static)
-  set_target_properties(png_static
+    add_library(${PROJECT_NAME} STATIC ${libpng_sources})
+    add_dependencies(${PROJECT_NAME} png_genfiles)
+    list(APPEND PNG_LIBRARY_TARGETS ${PROJECT_NAME})
+    set_target_properties(${PROJECT_NAME}
                         PROPERTIES OUTPUT_NAME "${PNG_STATIC_OUTPUT_NAME}"
                                    DEBUG_POSTFIX "${PNG_DEBUG_POSTFIX}")
-  target_include_directories(png_static
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
-  target_include_directories(png_static
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
-  target_include_directories(png_static
+  target_include_directories(${PROJECT_NAME}
                              SYSTEM
-                             INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/libpng${PNGLIB_ABI_VERSION}>")
-  target_link_libraries(png_static
-                        PUBLIC ${PNG_LINK_LIBRARIES})
-endif()
-
-if(PNG_FRAMEWORK AND NOT APPLE)
-  message(AUTHOR_WARNING "Setting PNG_FRAMEWORK to OFF, as it only applies to Apple systems")
-  set(PNG_FRAMEWORK OFF)
+                             INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+  target_link_libraries(${PROJECT_NAME}
+                        PUBLIC zlib ${M_LIBRARY})
 endif()
 
 if(PNG_FRAMEWORK)
-  add_library(png_framework SHARED ${libpng_sources})
-  add_dependencies(png_framework png_genfiles)
-  list(APPEND PNG_LIBRARY_TARGETS png_framework)
-  set_target_properties(png_framework
+    add_library(${PROJECT_NAME} SHARED ${libpng_sources})
+    add_dependencies(${PROJECT_NAME} png_genfiles)
+    list(APPEND PNG_LIBRARY_TARGETS ${PROJECT_NAME})
+    set_target_properties(${PROJECT_NAME}
                         PROPERTIES FRAMEWORK TRUE
                                    FRAMEWORK_VERSION "${PNGLIB_VERSION}"
                                    MACOSX_FRAMEWORK_SHORT_VERSION_STRING "${PNGLIB_MAJOR}.${PNGLIB_MINOR}"
                                    MACOSX_FRAMEWORK_BUNDLE_VERSION "${PNGLIB_VERSION}"
                                    MACOSX_FRAMEWORK_IDENTIFIER "org.libpng.libpng"
                                    XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
-                                   PUBLIC_HEADER "${libpng_public_hdrs}"
                                    OUTPUT_NAME "png"
                                    DEBUG_POSTFIX "${PNG_DEBUG_POSTFIX}")
   # Avoid CMake's implicit compile definition "-Dpng_framework_EXPORTS".
-  set_target_properties(png_framework
+  set_target_properties(${PROJECT_NAME}
                         PROPERTIES DEFINE_SYMBOL "")
-  target_include_directories(png_framework
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
-  target_include_directories(png_framework
+  target_include_directories(${PROJECT_NAME}
                              PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>")
-  target_include_directories(png_framework
+  target_include_directories(${PROJECT_NAME}
                              SYSTEM
-                             INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/libpng${PNGLIB_ABI_VERSION}>")
-  target_link_libraries(png_framework
-                        PUBLIC ${PNG_LINK_LIBRARIES})
+                             INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+  target_link_libraries(${PROJECT_NAME}
+                        PUBLIC zlib ${M_LIBRARY})
 endif()
 
 if(NOT PNG_LIBRARY_TARGETS)
@@ -774,7 +733,7 @@ if(PNG_TESTS AND PNG_SHARED)
 
   add_executable(pngtest ${pngtest_sources})
   target_link_libraries(pngtest
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
 
   png_add_test(NAME pngtest
                COMMAND pngtest
@@ -786,7 +745,7 @@ if(PNG_TESTS AND PNG_SHARED)
 
   add_executable(pngvalid ${pngvalid_sources})
   target_link_libraries(pngvalid
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
 
   png_add_test(NAME pngvalid-gamma-16-to-8
                COMMAND pngvalid
@@ -833,7 +792,7 @@ if(PNG_TESTS AND PNG_SHARED)
 
   add_executable(pngstest ${pngstest_sources})
   target_link_libraries(pngstest
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
 
   foreach(gamma_type 1.8 linear none sRGB)
     foreach(alpha_type none alpha)
@@ -889,7 +848,7 @@ if(PNG_TESTS AND PNG_SHARED)
 
   add_executable(pngunknown ${pngunknown_sources})
   target_link_libraries(pngunknown
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
 
   png_add_test(NAME pngunknown-discard
                COMMAND pngunknown
@@ -923,7 +882,7 @@ if(PNG_TESTS AND PNG_SHARED)
 
   add_executable(pngimage ${pngimage_sources})
   target_link_libraries(pngimage
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
 
   png_add_test(NAME pngimage-quick
                COMMAND pngimage
@@ -938,7 +897,7 @@ endif()
 if(PNG_SHARED AND PNG_TOOLS)
   add_executable(pngfix ${pngfix_sources})
   target_link_libraries(pngfix
-                        PRIVATE png_shared)
+                        PRIVATE ${PROJECT_NAME})
   set(PNG_BIN_TARGETS pngfix)
 
   add_executable(png-fix-itxt ${png_fix_itxt_sources})
@@ -1031,111 +990,20 @@ if(NOT WIN32 OR CYGWIN OR MINGW)
   set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
   set(LIBS "-lz -lm")
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in"
-                 "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
+                 "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
                  @ONLY)
-  create_symlink(libpng.pc FILE libpng${PNGLIB_ABI_VERSION}.pc)
+  create_symlink(libpng.pc FILE ${PROJECT_NAME}.pc)
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng-config.in"
-                 "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
+                 "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config"
                  @ONLY)
-  create_symlink(libpng-config FILE libpng${PNGLIB_ABI_VERSION}-config)
+  create_symlink(libpng-config FILE ${PROJECT_NAME}-config)
 endif()
 
-# Install.
-if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
-  install(TARGETS ${PNG_LIBRARY_TARGETS}
-          EXPORT libpng
-          RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-          LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-          ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-          FRAMEWORK DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-
-  if(PNG_SHARED)
-    # Create a symlink for libpng.dll.a => libpng16.dll.a on Cygwin
-    if(NOT WIN32 OR CYGWIN OR MINGW)
-      create_symlink(libpng${CMAKE_SHARED_LIBRARY_SUFFIX} TARGET png_shared)
-      install(FILES "$<TARGET_LINKER_FILE_DIR:png_shared>/libpng${CMAKE_SHARED_LIBRARY_SUFFIX}"
-              DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-    endif()
-  endif()
+# --- installation
 
-  if(PNG_STATIC)
-    if(NOT WIN32 OR CYGWIN OR MINGW)
-      create_symlink(libpng${CMAKE_STATIC_LIBRARY_SUFFIX} TARGET png_static)
-      install(FILES "$<TARGET_LINKER_FILE_DIR:png_static>/libpng${CMAKE_STATIC_LIBRARY_SUFFIX}"
-              DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-    endif()
-  endif()
-endif()
-
-if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL)
-  install(FILES ${libpng_public_hdrs}
-          DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
-  install(FILES ${libpng_public_hdrs}
-          DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/libpng${PNGLIB_ABI_VERSION}")
-endif()
-if(NOT SKIP_INSTALL_EXECUTABLES AND NOT SKIP_INSTALL_ALL)
-  if(NOT WIN32 OR CYGWIN OR MINGW)
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-  endif()
-endif()
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${libpng_public_hdrs}")
 
-if(NOT SKIP_INSTALL_PROGRAMS AND NOT SKIP_INSTALL_ALL)
-  install(TARGETS ${PNG_BIN_TARGETS}
-          RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
-endif()
-
-if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
-  # Install the man pages.
-  install(FILES libpng.3 libpngpf.3
-          DESTINATION "${CMAKE_INSTALL_MANDIR}/man3")
-  install(FILES png.5
-          DESTINATION "${CMAKE_INSTALL_MANDIR}/man5")
-  # Install the pkg-config files.
-  if(NOT CMAKE_HOST_WIN32 OR CYGWIN OR MINGW)
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng.pc"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-  endif()
-endif()
-
-# Create an export file that CMake users can include() to import our targets.
-if(NOT SKIP_INSTALL_EXPORT AND NOT SKIP_INSTALL_ALL)
-  install(EXPORT libpng
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/libpng"
-          FILE libpng${PNGLIB_ABI_VERSION}.cmake)
-endif()
-
-# Create a CMake Config File that can be used via find_package(PNG CONFIG)
-if(NOT SKIP_INSTALL_CONFIG_FILE AND NOT SKIP_INSTALL_ALL)
-  install(TARGETS ${PNG_LIBRARY_TARGETS}
-          EXPORT PNGTargets
-          RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-          LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-          ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-          FRAMEWORK DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-
-  include(CMakePackageConfigHelpers)
-  write_basic_package_version_file(PNGConfigVersion.cmake
-                                   VERSION ${PNGLIB_VERSION}
-                                   COMPATIBILITY SameMinorVersion)
-
-  install(EXPORT PNGTargets
-          FILE PNGTargets.cmake
-          NAMESPACE PNG::
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/PNG")
-
-  install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/scripts/cmake/PNGConfig.cmake"
-                "${CMAKE_CURRENT_BINARY_DIR}/PNGConfigVersion.cmake"
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/PNG")
-endif()
+include("${CMAKE_CURRENT_SOURCE_DIR}/Installing.cmake")
 
 # TODO: Create MSVC import lib for MinGW-compiled shared lib.
 # pexports libpng.dll > libpng.def
diff --git a/pngstruct.h b/pngstruct.h
index fe5fa0415..97a267437 100644
--- a/pngstruct.h
+++ b/pngstruct.h
@@ -24,7 +24,7 @@
    /* We must ensure that zlib uses 'const' in declarations. */
 #  define ZLIB_CONST
 #endif
-#include "zlib.h"
+#include <zlib/zlib.h>
 #ifdef const
    /* zlib.h sometimes #defines const to nothing, undo this. */
 #  undef const
diff --git a/pngtest.c b/pngtest.c
index 08140dd44..eb7064fba 100644
--- a/pngtest.c
+++ b/pngtest.c
@@ -39,7 +39,7 @@
 #ifdef PNG_ZLIB_HEADER
 #  include PNG_ZLIB_HEADER /* defined by pnglibconf.h from 1.7 */
 #else
-#  include <zlib.h>
+#  include <zlib/zlib.h>
 #endif
 
 #include "png.h"
diff --git a/scripts/pnglibconf.dfa b/scripts/pnglibconf.dfa
index f466da1a3..ce33dea69 100644
--- a/scripts/pnglibconf.dfa
+++ b/scripts/pnglibconf.dfa
@@ -343,7 +343,7 @@ option MIPS_MMI_CHECK disabled requires ALIGNED_MEMORY,
 # The TEXT values are the defaults when writing compressed text (all forms)
 
 # Include the zlib header so that the defaults below are known
-@#  include <zlib.h>
+@#  include <zlib/zlib.h>
 
 # The '@' here means to substitute the value when pnglibconf.h is built
 setting Z_DEFAULT_COMPRESSION default @Z_DEFAULT_COMPRESSION
