diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7711c7..f05de0b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,14 @@
 # SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
 #    https://www.khronos.org/registry/
 
-cmake_minimum_required(VERSION 3.14)
-project(SPIRV-Headers LANGUAGES C CXX VERSION 1.5.5)
+cmake_minimum_required(VERSION 3.22)
+
+set(SPIRV_HEADERS_VERSION "1.5.5" CACHE STRING "SPIRV-Headers version")
+
+project(SPIRV-Headers
+    LANGUAGES C CXX
+    VERSION "${SPIRV_HEADERS_VERSION}"
+)
 
 if (CMAKE_VERSION VERSION_LESS "3.21")
     # https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html
