diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cfba73..1abfef2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,8 +28,8 @@
 # The SPIR-V headers from the SPIR-V Registry
 # https://www.khronos.org/registry/spir-v/
 #
-cmake_minimum_required(VERSION 3.0)
-project(SPIRV-Headers VERSION 1.5.5)
+cmake_minimum_required(VERSION 3.22)
+project(SPIRV-Headers VERSION 2023.3.8)
 
 # There are two ways to use this project.
 #
