diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a168db125..53185d548 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1310,11 +1310,12 @@ else()
 endif()
 set_target_properties(xerces-c PROPERTIES FOLDER "Library")
 
-if(WIN32)
-  set(xerces_config_dir "cmake")
-else()
-  set(xerces_config_dir "${CMAKE_INSTALL_LIBDIR}/cmake/XercesC")
-endif()
+# if(WIN32)
+#   set(xerces_config_dir "cmake")
+# else()
+#   set(xerces_config_dir "${CMAKE_INSTALL_LIBDIR}/cmake/XercesC")
+# endif()
+set(xerces_config_dir "${CMAKE_INSTALL_LIBDIR}/cmake/XercesC")
 
 install(TARGETS xerces-c
   EXPORT XercesCConfigInternal
