diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7009ba47a..9b4815029a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,27 +26,28 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 # As per https://github.com/google/oss-policies-info/blob/main/foundational-cxx-support-matrix.md
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.28)
 
 # - Since we are past CMake 3.12 we can add CONFIGURE_DEPENDS to DawnGenerator to rerun CMake
 #   in case any of the generator files changes. We should also remove the CACHE "" FORCE stuff to
 #   override options in third_party dependencies.
 
+set(DAWN_VERSION "0.0.0" CACHE STRING "Dawn version")
+
 project(
     Dawn
     DESCRIPTION "Dawn, a WebGPU implementation"
     LANGUAGES C CXX
     HOMEPAGE_URL "https://dawn.googlesource.com/dawn"
-    VERSION 0.0.0
+    VERSION ${DAWN_VERSION}
 )
-enable_testing()
 
 list(INSERT CMAKE_MODULE_PATH 0 "${Dawn_SOURCE_DIR}/src/cmake")
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
-set(CMAKE_DEBUG_POSTFIX "")
+set(CMAKE_DEBUG_POSTFIX "d" CACHE STRING "Filename postfix for libraries under DEBUG configuration")
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 set(IS_DEBUG_BUILD 0)
@@ -132,7 +133,7 @@ if (DAWN_SUPPORTS_GLFW_FOR_WINDOWING)
 endif()
 
 option(DAWN_ENABLE_ASAN "Enable address sanitizer" OFF)
-option(DAWN_ENABLE_INSTALL "Enable install step for Dawn libraries" OFF)
+option(DAWN_ENABLE_INSTALL "Enable install step for Dawn libraries" ON)
 option(DAWN_ENABLE_TSAN "Enable thread sanitizer" OFF)
 option(DAWN_ENABLE_MSAN "Enable memory sanitizer" OFF)
 option(DAWN_ENABLE_UBSAN "Enable undefined behaviour sanitizer" OFF)
@@ -180,30 +181,20 @@ set_if_not_defined(LLVM_SOURCE_DIR "${Dawn_LLVM_SOURCE_DIR}" "Directory to an LL
 
 set_if_not_defined(DAWN_THIRD_PARTY_DIR "${Dawn_SOURCE_DIR}/third_party" "Directory in which to find third-party dependencies.")
 
-set_if_not_defined(DAWN_ABSEIL_DIR "${DAWN_THIRD_PARTY_DIR}/abseil-cpp" "Directory in which to find Abseil")
-set_if_not_defined(DAWN_GLFW_DIR "${DAWN_THIRD_PARTY_DIR}/glfw3/src" "Directory in which to find GLFW")
-set_if_not_defined(DAWN_JINJA2_DIR "${DAWN_THIRD_PARTY_DIR}/jinja2" "Directory in which to find Jinja2")
-set_if_not_defined(DAWN_MARKUPSAFE_DIR "${DAWN_THIRD_PARTY_DIR}/markupsafe" "Directory in which to find MarkupSafe")
-set_if_not_defined(DAWN_KHRONOS_DIR "${DAWN_THIRD_PARTY_DIR}/khronos" "Directory in which to find Khronos GL headers")
-set_if_not_defined(DAWN_SWIFTSHADER_DIR "${DAWN_THIRD_PARTY_DIR}/swiftshader" "Directory in which to find swiftshader")
-set_if_not_defined(DAWN_PROTOBUF_DIR "${DAWN_THIRD_PARTY_DIR}/protobuf" "Directory in which to find protobuf")
-set_if_not_defined(DAWN_LPM_DIR "${DAWN_THIRD_PARTY_DIR}/libprotobuf-mutator/src" "Directory in which to find libprotobuf")
 set_if_not_defined(DAWN_EMDAWNWEBGPU_DIR "${DAWN_THIRD_PARTY_DIR}/emdawnwebgpu" "Directory in which to find Dawn specific Emscripten bindings")
-set_if_not_defined(DAWN_GOOGLETEST_DIR "${DAWN_THIRD_PARTY_DIR}/googletest" "Directory in which to find googletest")
-set_if_not_defined(DAWN_EGL_REGISTRY_DIR "${DAWN_THIRD_PARTY_DIR}/EGL-Registry/src" "Directory in which to find EGL-Registry")
-set_if_not_defined(DAWN_OPENGL_REGISTRY_DIR "${DAWN_THIRD_PARTY_DIR}/OpenGL-Registry/src" "Directory in which to find OpenGL-Registry")
-set_if_not_defined(DAWN_SPIRV_TOOLS_DIR "${DAWN_THIRD_PARTY_DIR}/spirv-tools/src" "Directory in which to find SPIRV-Tools")
-set_if_not_defined(DAWN_SPIRV_HEADERS_DIR "${DAWN_THIRD_PARTY_DIR}/spirv-headers/src" "Directory in which to find SPIRV-Headers")
-set_if_not_defined(DAWN_GLSLANG_DIR "${DAWN_THIRD_PARTY_DIR}/glslang/src" "Directory in which to find GLSLang")
-set_if_not_defined(DAWN_VULKAN_HEADERS_DIR "${DAWN_THIRD_PARTY_DIR}/vulkan-headers/src" "Directory in which to find Vulkan-Headers")
-set_if_not_defined(DAWN_VULKAN_LOADER_DIR "${DAWN_THIRD_PARTY_DIR}/vulkan-loader/src" "Directory in which to find Vulkan-Loader")
-set_if_not_defined(DAWN_VULKAN_UTILITY_LIBRARIES_DIR "${DAWN_THIRD_PARTY_DIR}/vulkan-utility-libraries/src" "Directory in which to find Vulkan-Utility-Libraries")
-set_if_not_defined(DAWN_WEBGPU_HEADERS_DIR "${DAWN_THIRD_PARTY_DIR}/webgpu-headers/src" "Directory in which to find webgpu headers")
-
 
 # Handling of the type of monolithic library to build, if any.
 set(DAWN_BUILD_MONOLITHIC_LIBRARY "STATIC" CACHE STRING "Build monolithic library: SHARED, STATIC, or OFF.")
-set_property(CACHE DAWN_BUILD_MONOLITHIC_LIBRARY PROPERTY STRINGS SHARED SHARED STATIC)
+set_property(CACHE DAWN_BUILD_MONOLITHIC_LIBRARY PROPERTY STRINGS OFF SHARED STATIC)
+# this combination doesn't seem to be supported, as it will error about exporting DawnTargets
+if(EMSCRIPTEN AND DAWN_BUILD_MONOLITHIC_LIBRARY)
+    message(
+        WARNING
+            "Building with Emscripten, but DAWN_BUILD_MONOLITHIC_LIBRARY is set "
+            "to ${DAWN_BUILD_MONOLITHIC_LIBRARY} - will set it to OFF"
+    )
+    set(DAWN_BUILD_MONOLITHIC_LIBRARY "OFF")
+endif()
 
 string(TOUPPER "${DAWN_BUILD_MONOLITHIC_LIBRARY}" _option_upper)
 if(NOT _option_upper STREQUAL "OFF" AND
@@ -211,9 +202,10 @@ if(NOT _option_upper STREQUAL "OFF" AND
    NOT _option_upper STREQUAL "STATIC")
     message(FATAL_ERROR "DAWN_BUILD_MONOLITHIC_LIBRARY must be SHARED, STATIC, or OFF, but was \"${DAWN_BUILD_MONOLITHIC_LIBRARY}\".")
 endif()
-if (DAWN_BUILD_MONOLITHIC_LIBRARY AND BUILD_SHARED_LIBS)
-    message(FATAL_ERROR "DAWN_BUILD_MONOLITHIC_LIBRARY SHARED/STATIC requires BUILD_SHARED_LIBS=OFF. Otherwise the bundled library will depend on other shared libraries which defeats the purpose.")
-endif()
+# what the fuck is this, how then one is supposed to build it as SHARED with BUILD_SHARED_LIBS
+#if (DAWN_BUILD_MONOLITHIC_LIBRARY AND BUILD_SHARED_LIBS)
+#    message(FATAL_ERROR "DAWN_BUILD_MONOLITHIC_LIBRARY SHARED/STATIC requires BUILD_SHARED_LIBS=OFF. Otherwise the bundled library will depend on other shared libraries which defeats the purpose.")
+#endif()
 option(DAWN_FETCH_DEPENDENCIES "Use fetch_dawn_dependencies.py as an alternative to using depot_tools" OFF)
 
 if (DAWN_ENABLE_OPENGLES OR DAWN_ENABLE_DESKTOP_GL)
@@ -264,12 +256,6 @@ option(TINT_ENABLE_BREAK_IN_DEBUGGER "Enable tint::debugger::Break()" OFF)
 option(TINT_CHECK_CHROMIUM_STYLE "Check for [chromium-style] issues during build" OFF)
 option(TINT_RANDOMIZE_HASHES "Randomize the hash seed value to detect non-deterministic output" OFF)
 
-# Dependencies for DAWN_BUILD_NODE_BINDINGS
-set_if_not_defined(NODE_ADDON_API_DIR "${DAWN_THIRD_PARTY_DIR}/node-addon-api" "Directory in which to find node-addon-api")
-set_if_not_defined(NODE_API_HEADERS_DIR "${DAWN_THIRD_PARTY_DIR}/node-api-headers" "Directory in which to find node-api-headers")
-set_if_not_defined(WEBGPU_IDL_PATH "${DAWN_THIRD_PARTY_DIR}/gpuweb/webgpu.idl" "Path to the webgpu.idl definition file")
-set_if_not_defined(GO_EXECUTABLE "go" "Golang executable for running the IDL generator")
-
 message(STATUS "Build type...........: ${CMAKE_BUILD_TYPE}")
 message(STATUS "C++ compiler.........: ${CMAKE_CXX_COMPILER}")
 message(STATUS "CMake generator......: ${CMAKE_GENERATOR}")
@@ -342,28 +328,6 @@ message(STATUS "Tint randomize hashes: ${TINT_RANDOMIZE_HASHES}")
 message(STATUS "Tint build Mesa: ${TINT_BUILD_MESA}")
 message(STATUS "")
 
-message(STATUS "Dawn third_party dir: ${DAWN_THIRD_PARTY_DIR}")
-message(STATUS "Dawn GLFW dir: ${DAWN_GLFW_DIR}")
-message(STATUS "Dawn Jinja2 dir: ${DAWN_JINJA2_DIR}")
-message(STATUS "Dawn MarkupSafe dir: ${DAWN_MARKUPSAFE_DIR}")
-message(STATUS "Dawn Khronos dir: ${DAWN_KHRONOS_DIR}")
-message(STATUS "Dawn Swiftshader dir: ${DAWN_SWIFTSHADER_DIR}")
-message(STATUS "Dawn Protobuf dir: ${DAWN_PROTOBUF_DIR}")
-message(STATUS "Dawn LPM dir: ${DAWN_LPM_DIR}")
-message(STATUS "Dawn Emdawnwebgpu dir: ${DAWN_EMDAWNWEBGPU_DIR}")
-message(STATUS "Dawn Spir-Tools dir: ${DAWN_SPIRV_TOOLS_DIR}")
-message(STATUS "Dawn Spirv-Headers dir: ${DAWN_SPIRV_HEADERS_DIR}")
-message(STATUS "Dawn Glslang dir: ${DAWN_GLSLANG_DIR}")
-message(STATUS "Dawn Vulkan Headers dir: ${DAWN_VULKAN_HEADERS_DIR}")
-message(STATUS "Dawn Vulkan Utility Libraries dir: ${DAWN_VULKAN_UTILITY_LIBRARIES_DIR}")
-message(STATUS "")
-
-message(STATUS "Node Addon API dir: ${NODE_ADDON_API_DIR}")
-message(STATUS "Node API Headers dir: ${NODE_API_HEADERS_DIR}")
-message(STATUS "Webgpu IDL path: ${WEBGPU_IDL_PATH}")
-message(STATUS "Go exe: ${GO_EXECUTABLE}")
-message(STATUS "")
-
 if (TINT_BUILD_FUZZER_VULKAN_SUPPORT AND NOT TINT_BUILD_FUZZERS)
     message(FATAL_ERROR "Cannot enable TINT_BUILD_FUZZER_VULKAN_SUPPORT without enabling TINT_BUILD_FUZZERS")
 endif()
@@ -390,9 +354,9 @@ target_include_directories(dawn_public_config INTERFACE
 # The internal config contains additional path but includes the dawn_public_config include paths
 add_library(dawn_internal_config INTERFACE)
 target_include_directories(dawn_internal_config INTERFACE
-    "${PROJECT_SOURCE_DIR}"
-    "${DAWN_SRC_DIR}"
-    "${DAWN_BUILD_GEN_DIR}/src"
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>"
+    "$<BUILD_INTERFACE:${DAWN_SRC_DIR}>"
+    "$<BUILD_INTERFACE:${DAWN_BUILD_GEN_DIR}/src>"
 )
 target_link_libraries(dawn_internal_config INTERFACE dawn_public_config)
 
@@ -468,24 +432,57 @@ endif()
 set(TINT_LIB_FUZZING_ENGINE_LINK_OPTIONS "" CACHE STRING "Used by OSS-Fuzz to control, via link options, which fuzzing engine should be used")
 
 set(TINT_ROOT_SOURCE_DIR   ${PROJECT_SOURCE_DIR})
-set(TINT_SPIRV_HEADERS_DIR ${DAWN_SPIRV_HEADERS_DIR})
-set(TINT_SPIRV_TOOLS_DIR   ${DAWN_SPIRV_TOOLS_DIR})
 
+find_package(absl CONFIG REQUIRED)
+
+if(DAWN_USE_GLFW)
+    find_package(glfw3 CONFIG REQUIRED)
+endif()
+
+if(TINT_BUILD_SPV_READER OR TINT_BUILD_SPV_WRITER)
+    find_package(SPIRV-Headers CONFIG REQUIRED)
+    find_package(SPIRV-Tools CONFIG REQUIRED)
+    find_package(SPIRV-Tools-opt CONFIG REQUIRED)
+endif()
+
+if (DAWN_ENABLE_VULKAN)
+    find_package(VulkanHeaders CONFIG REQUIRED)
+    find_package(VulkanUtilityLibraries CONFIG REQUIRED)
+endif()
 
 ################################################################################
 # Run on all subdirectories
 ################################################################################
-add_subdirectory(third_party)
+
+# don't vendor dependencies
+#add_subdirectory(third_party)
+
+function(add_includes_list HEADERS ROOT_DIR)
+    foreach(header ${HEADERS})
+        file(TO_CMAKE_PATH "${header}" FIXED_HEADER)
+        file(RELATIVE_PATH header_file_path "${ROOT_DIR}" "${FIXED_HEADER}")
+        get_filename_component(header_directory_path "${header_file_path}" DIRECTORY)
+        install(
+            FILES ${FIXED_HEADER}
+            DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${header_directory_path}"
+        )
+    endforeach()
+endfunction()
+
+function(add_includes_glob GLOB ROOT_DIR)
+    file(GLOB_RECURSE public_headers ${GLOB})
+    add_includes_list("${public_headers}" "${ROOT_DIR}")
+endfunction()
 
 # TODO(crbug.com/tint/455): Tint does not currently build with CMake when
 # BUILD_SHARED_LIBS=1, so always build it as static for now.
-set(BUILD_SHARED_LIBS_SAVED ${BUILD_SHARED_LIBS})
-set(BUILD_SHARED_LIBS 0)
+#set(BUILD_SHARED_LIBS_SAVED ${BUILD_SHARED_LIBS})
+#set(BUILD_SHARED_LIBS 0)
 
 add_subdirectory(src/utils)
 add_subdirectory(src/tint)
 
-set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS_SAVED})
+#set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS_SAVED})
 
 if (DAWN_ENABLE_D3D11 OR DAWN_ENABLE_D3D12 OR DAWN_ENABLE_METAL OR DAWN_ENABLE_NULL OR DAWN_ENABLE_DESKTOP_GL OR DAWN_ENABLE_OPENGLES OR DAWN_ENABLE_VULKAN OR EMSCRIPTEN)
     add_subdirectory(generator)
@@ -529,13 +526,14 @@ if (DAWN_EMIT_COVERAGE)
     VERBATIM)
 endif()
 
-if (DAWN_BUILD_MONOLITHIC_LIBRARY AND DAWN_ENABLE_INSTALL)
+# no idea what purpose does this "monolithic" library serve
+if(DAWN_BUILD_MONOLITHIC_LIBRARY) # any non-"OFF" value
   # This series of functions add the necessary information so that other CMake projects
   # can use Dawn, be it from a build directory, a local install or when packaged.
   install(EXPORT DawnTargets
     FILE DawnTargets.cmake
     NAMESPACE dawn::
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Dawn")
+    DESTINATION "share/${PROJECT_NAME}")
   export(EXPORT DawnTargets
     FILE "${CMAKE_CURRENT_BINARY_DIR}/cmake/DawnTargets.cmake"
       NAMESPACE dawn::
@@ -548,12 +546,278 @@ if (DAWN_BUILD_MONOLITHIC_LIBRARY AND DAWN_ENABLE_INSTALL)
   # Configure config file
   configure_package_config_file("${CMAKE_CURRENT_LIST_DIR}/src/cmake/DawnConfig.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/DawnConfig.cmake"
-    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Dawn"
+    INSTALL_DESTINATION "share/${PROJECT_NAME}"
   )
   # Install the fully generated config and configVersion files
   install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/DawnConfig.cmake"
     "${CMAKE_CURRENT_BINARY_DIR}/DawnConfigVersion.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Dawn"
+    DESTINATION "share/${PROJECT_NAME}"
   )
-endif ()
+else()
+    # trying to figure out which targets are supposed to be installed
+    #
+    set(TINT_COMMON_TARGETS "")
+    if(TINT_BUILD_MSL_WRITER
+        OR (TINT_BUILD_SPV_READER OR TINT_BUILD_SPV_WRITER)
+        # OR (TINT_BUILD_WGSL_READER OR TINT_BUILD_WGSL_WRITER)
+        # OR SOME_OTHER?
+    )
+        list(APPEND TINT_COMMON_TARGETS
+            tint_lang_core_ir_transform
+        )
+    endif()
+    #
+    set(TINT_GLSL_TARGETS "")
+    if(TINT_BUILD_GLSL_WRITER)
+        list(APPEND TINT_GLSL_TARGETS
+            tint_lang_glsl
+            tint_lang_glsl_ir
+            tint_lang_glsl_intrinsic
+            tint_lang_glsl_writer
+            #tint_lang_glsl_writer_ast_printer
+            #tint_lang_glsl_writer_ast_raise
+            tint_lang_glsl_writer_common
+            tint_lang_glsl_writer_printer
+            tint_lang_glsl_writer_raise
+        )
+    endif()
+    #
+    set(TINT_HLSL_TARGETS "")
+    if(TINT_BUILD_HLSL_WRITER)
+        list(APPEND TINT_HLSL_TARGETS
+            tint_lang_hlsl
+            tint_lang_hlsl_intrinsic
+            tint_lang_hlsl_ir
+            tint_lang_hlsl_type
+            tint_lang_hlsl_writer
+            tint_lang_hlsl_writer_printer
+            tint_lang_hlsl_writer_raise
+            #tint_lang_hlsl_writer_ast_printer
+            #tint_lang_hlsl_writer_ast_raise
+        )
+    endif()
+    #
+    set(TINT_MSL_TARGETS "")
+    if(TINT_BUILD_MSL_WRITER)
+        list(APPEND TINT_MSL_TARGETS
+            tint_lang_msl
+            tint_lang_msl_intrinsic
+            tint_lang_msl_ir
+            tint_lang_msl_type
+            tint_lang_msl_writer
+            #tint_lang_msl_writer_ast_printer
+            #tint_lang_msl_writer_ast_raise
+            tint_lang_msl_writer_common
+            tint_lang_msl_writer_printer
+            tint_lang_msl_writer_raise
+        )
+    endif()
+    #
+    set(TINT_SPIRV_TARGETS "")
+    if(TINT_BUILD_SPV_READER OR TINT_BUILD_SPV_WRITER)
+        list(APPEND TINT_SPIRV_TARGETS
+            tint_lang_spirv
+            tint_lang_spirv_intrinsic
+            tint_lang_spirv_ir
+            tint_lang_spirv_type
+            tint_lang_spirv_validate
+        )
+    endif()
+    if(TINT_BUILD_SPV_READER)
+        list(APPEND TINT_SPIRV_TARGETS
+            tint_lang_spirv_reader
+            #tint_lang_spirv_reader_ast_lower
+            #tint_lang_spirv_reader_ast_parser
+            tint_lang_spirv_reader_parser
+        )
+    endif()
+    if(TINT_BUILD_SPV_WRITER)
+        list(APPEND TINT_SPIRV_TARGETS
+            tint_lang_spirv_writer
+            #tint_lang_spirv_writer_ast_printer
+            #tint_lang_spirv_writer_ast_raise
+            tint_lang_spirv_writer_common
+            tint_lang_spirv_writer_printer
+            tint_lang_spirv_writer_raise
+        )
+    endif()
+    #
+    set(TINT_WGSL_TARGETS "")
+    if(TINT_BUILD_WGSL_READER OR TINT_BUILD_WGSL_WRITER)
+        list(APPEND TINT_WGSL_TARGETS
+            #tint_lang_wgsl_ast_transform
+            #tint_lang_wgsl_helpers
+            tint_lang_wgsl_inspector
+        )
+    endif()
+    if(TINT_BUILD_WGSL_READER)
+        list(APPEND TINT_WGSL_TARGETS
+            tint_lang_wgsl_reader
+            tint_lang_wgsl_reader_lower
+            tint_lang_wgsl_reader_parser
+            tint_lang_wgsl_reader_program_to_ir
+        )
+    endif()
+    if(TINT_BUILD_WGSL_WRITER)
+        list(APPEND TINT_WGSL_TARGETS
+            tint_lang_wgsl_writer
+            #tint_lang_wgsl_writer_ast_printer
+            tint_lang_wgsl_writer_raise
+            #tint_lang_wgsl_writer_syntax_tree_printer
+        )
+    endif()
+    #
+    set(DAWN_NON_EMSCRIPTEN_TARGETS "")
+    set(DAWN_EMSCRIPTEN_TARGETS "")
+    if(NOT EMSCRIPTEN) # non-Emscripten targets
+        list(APPEND DAWN_NON_EMSCRIPTEN_TARGETS
+            dawn_headers
+            dawn_native
+            dawn_native_objects
+            dawn_platform
+            dawn_proc
+            dawn_wire
+            dawncpp_headers
+            tint_lang_null_writer
+            tint_lang_null_writer_raise
+        )
+    else() # Emscripten targets
+        list(APPEND DAWN_EMSCRIPTEN_TARGETS
+            emdawnwebgpu_c
+            emdawnwebgpu_c_include
+            emdawnwebgpu_cpp
+        )
+    endif()
+    #
+    install( # a lot of these are internal targets, which actually shouldn't be exposed
+        TARGETS
+            ${DAWN_NON_EMSCRIPTEN_TARGETS}
+            ${DAWN_EMSCRIPTEN_TARGETS}
+            dawn_common
+            dawn_internal_config
+            dawn_public_config
+            dawn_shared_utils
+            dawn_system_utils
+            partition_alloc
+            tint_api
+            tint_api_common
+            #tint_api_options
+            ${TINT_COMMON_TARGETS}
+            tint_lang_core
+            #tint_lang_core_common
+            tint_lang_core_constant
+            tint_lang_core_intrinsic
+            tint_lang_core_ir
+            tint_lang_core_ir_analysis
+            tint_lang_core_ir_type
+            tint_lang_core_type
+            ${TINT_GLSL_TARGETS}
+            ${TINT_HLSL_TARGETS}
+            tint_lang_hlsl_writer_common
+            ${TINT_MSL_TARGETS}
+            tint_lang_null_writer_common
+            ${TINT_SPIRV_TARGETS}
+            tint_lang_spirv_reader_common
+            tint_lang_spirv_reader_lower
+            tint_lang_spirv_writer_analysis
+            ${TINT_WGSL_TARGETS}
+            tint_lang_wgsl
+            tint_lang_wgsl_ast
+            #tint_lang_wgsl_common
+            #tint_lang_wgsl_features
+            tint_lang_wgsl_intrinsic
+            tint_lang_wgsl_ir
+            tint_lang_wgsl_program
+            tint_lang_wgsl_resolver
+            tint_lang_wgsl_sem
+            tint_lang_wgsl_writer_ast_printer
+            tint_lang_wgsl_writer_common
+            tint_lang_wgsl_writer_ir_to_program
+            tint_utils
+            tint_utils_containers
+            tint_utils_text_generator
+            #tint_utils_debug
+            tint_utils_diagnostic
+            #tint_utils_generator
+            tint_utils_ice
+            #tint_utils_id
+            tint_utils_macros
+            tint_utils_math
+            tint_utils_memory
+            #tint_utils_reflection
+            #tint_utils_result
+            tint_utils_rtti
+            tint_utils_strconv
+            tint_utils_symbol
+            tint_utils_system
+            tint_utils_text
+            #tint_utils_traits
+        EXPORT "${PROJECT_NAME}Targets"
+        PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/$<LOWER_CASE:${PROJECT_NAME}>"
+        INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+    )
+    if(DAWN_USE_GLFW)
+        install(TARGETS dawn_glfw
+            EXPORT "${PROJECT_NAME}Targets"
+            PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/$<LOWER_CASE:${PROJECT_NAME}>"
+            INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+        )
+    endif()
+    # generate and install export file
+    install(EXPORT "${PROJECT_NAME}Targets"
+        FILE "${PROJECT_NAME}Targets.cmake"
+        NAMESPACE "dawn::"
+        DESTINATION "share/${PROJECT_NAME}"
+    )
+    add_includes_glob(
+        "${PROJECT_SOURCE_DIR}/include/*.h"
+        "${CMAKE_CURRENT_SOURCE_DIR}/include"
+    )
+    #add_includes_glob(
+    #    "${PROJECT_BINARY_DIR}/gen/include/*.h"
+    #    "${CMAKE_CURRENT_BINARY_DIR}/gen/include"
+    #)
+    if(NOT EMSCRIPTEN)
+        install(
+            DIRECTORY
+                "${PROJECT_BINARY_DIR}/gen/include/"
+            DESTINATION
+                "${CMAKE_INSTALL_INCLUDEDIR}"
+        )
+    endif()
+    install(
+        FILES
+            "${PROJECT_SOURCE_DIR}/src/utils/compiler.h"
+            "${PROJECT_SOURCE_DIR}/src/utils/numeric.h"
+        DESTINATION
+            "${CMAKE_INSTALL_INCLUDEDIR}/$<LOWER_CASE:${PROJECT_NAME}/utils>"
+    )
+    add_includes_glob(
+        "${PROJECT_SOURCE_DIR}/src/tint/*.h"
+        "${CMAKE_CURRENT_SOURCE_DIR}/src"
+    )
+    include(CMakePackageConfigHelpers)
+    configure_package_config_file(
+        "${CMAKE_CURRENT_LIST_DIR}/src/cmake/DawnConfig.cmake.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+        INSTALL_DESTINATION "share/${PROJECT_NAME}"
+    )
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+        DESTINATION "share/${PROJECT_NAME}"
+    )
+    # all of these are surprisingly required to be delivered to consuming projects too,
+    # because they are referenced in the resulting DawnTargets.cmake in target_link_options()
+    # with absolute paths
+    if(EMSCRIPTEN)
+        install(
+            FILES
+                "${CMAKE_CURRENT_BINARY_DIR}/gen/src/emdawnwebgpu/library_webgpu_enum_tables.js"
+                "${CMAKE_CURRENT_BINARY_DIR}/gen/src/emdawnwebgpu/library_webgpu_generated_sig_info.js"
+                "${CMAKE_CURRENT_BINARY_DIR}/gen/src/emdawnwebgpu/library_webgpu_generated_struct_info.js"
+                "${PROJECT_SOURCE_DIR}/third_party/emdawnwebgpu/pkg/webgpu/src/library_webgpu.js"
+                "${PROJECT_SOURCE_DIR}/third_party/emdawnwebgpu/pkg/webgpu/src/webgpu-externs.js"
+            DESTINATION "share/${PROJECT_NAME}/js"
+        )
+    endif()
+endif()
diff --git a/generator/CMakeLists.txt b/generator/CMakeLists.txt
index 67febfea5a..47cdfe26de 100644
--- a/generator/CMakeLists.txt
+++ b/generator/CMakeLists.txt
@@ -118,11 +118,15 @@ function(DawnGenerator)
     foreach (filename IN LISTS OUTPUTS)
         get_filename_component(extension "${filename}" EXT)
         if (extension MATCHES "(h|hpp)")
-            list(APPEND headers "${filename}")
+            if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+                list(APPEND headers "${filename}")
+            else()
+                list(APPEND headers "$<BUILD_INTERFACE:${filename}>")
+            endif()
         elseif (extension MATCHES "(json)")
             list(APPEND jsons "${filename}")
         elseif (extension MATCHES "(js)")
-            list(APPEND js "${filename}")
+            list(APPEND js "$<BUILD_INTERFACE:${filename}>")
         else ()
             list(APPEND sources "${filename}")
         endif ()
diff --git a/src/cmake/DawnConfig.cmake.in b/src/cmake/DawnConfig.cmake.in
index 8bfeda55bc..69b1f60ee8 100644
--- a/src/cmake/DawnConfig.cmake.in
+++ b/src/cmake/DawnConfig.cmake.in
@@ -31,4 +31,28 @@
 # 2. https://cmake.org/cmake/help/latest/guide/tutorial/Adding%20Export%20Configuration.html
 
 @PACKAGE_INIT@
-include ("${CMAKE_CURRENT_LIST_DIR}/DawnTargets.cmake")
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(absl CONFIG REQUIRED)
+
+if(@DAWN_USE_GLFW@)
+    find_dependency(glfw3 CONFIG REQUIRED)
+endif()
+
+#if("@CMAKE_SYSTEM_NAME@" STREQUAL "Windows")
+if(@TINT_BUILD_SPV_READER@ OR @TINT_BUILD_SPV_WRITER@)
+    find_dependency(SPIRV-Headers CONFIG REQUIRED)
+    find_dependency(SPIRV-Tools CONFIG REQUIRED)
+    find_dependency(SPIRV-Tools-opt CONFIG REQUIRED)
+    # some other SPIRV targets?
+endif()
+
+if(@DAWN_ENABLE_VULKAN@)
+    find_dependency(VulkanHeaders CONFIG REQUIRED)
+    find_dependency(VulkanUtilityLibraries CONFIG REQUIRED)
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
+
+check_required_components(@PROJECT_NAME@)
diff --git a/src/dawn/CMakeLists.txt b/src/dawn/CMakeLists.txt
index c0de9721af..3503854a64 100644
--- a/src/dawn/CMakeLists.txt
+++ b/src/dawn/CMakeLists.txt
@@ -35,16 +35,26 @@ DawnJSONGenerator(
     OUTPUT_HEADERS DAWN_HEADERS_GEN_HEADERS
 )
 
+set(DAWN_HEADERS_LIST
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu.h>"
+    ${DAWN_HEADERS_GEN_HEADERS}
+)
+if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+    set(DAWN_HEADERS_LIST
+        "${DAWN_INCLUDE_DIR}/webgpu/webgpu.h"
+        ${DAWN_HEADERS_GEN_HEADERS}
+    )
+endif()
 dawn_add_library(
     dawn_headers
     HEADER_ONLY
     UTILITY_TARGET dawn_internal_config
     HEADERS
-        "${DAWN_INCLUDE_DIR}/webgpu/webgpu.h"
-        ${DAWN_HEADERS_GEN_HEADERS}
+        "${DAWN_HEADERS_LIST}"
     DEPENDS
         dawn_public_config
 )
+unset(DAWN_HEADERS_LIST)
 
 if(BUILD_SHARED_LIBS)
   target_compile_definitions(dawn_headers INTERFACE "WGPU_SHARED_LIBRARY")
@@ -65,18 +75,30 @@ DawnJSONGenerator(
     PRINT_NAME "Dawn C++ headers"
     OUTPUT_HEADERS DAWNCPP_HEADERS_GEN_HEADERS
 )
+set(DAWNCPP_HEADERS_LIST
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp_print.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu_enum_class_bitmasks.h>"
+    ${DAWNCPP_HEADERS_GEN_HEADERS}
+)
+if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+    set(DAWNCPP_HEADERS_LIST
+        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp.h"
+        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp_print.h"
+        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_enum_class_bitmasks.h"
+        ${DAWNCPP_HEADERS_GEN_HEADERS}
+    )
+endif()
 dawn_add_library(
     dawncpp_headers
     HEADER_ONLY
     UTILITY_TARGET dawn_internal_config
     HEADERS
-        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp.h"
-        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_cpp_print.h"
-        "${DAWN_INCLUDE_DIR}/webgpu/webgpu_enum_class_bitmasks.h"
-        ${DAWNCPP_HEADERS_GEN_HEADERS}
+        "${DAWNCPP_HEADERS_LIST}"
     DEPENDS
         dawn::dawn_headers
 )
+unset(DAWNCPP_HEADERS_LIST)
 
 if (EMSCRIPTEN)
   add_library(webgpu_cpp ALIAS emdawnwebgpu_cpp)
@@ -128,12 +150,21 @@ if (NOT EMSCRIPTEN)
       OUTPUT_SOURCES DAWNPROC_GEN_SOURCES
   )
 
+  set(DAWN_PROC_HEADERS_LIST
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/dawn_thread_dispatch_proc.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/dawn_proc.h>"
+  )
+  if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+    set(DAWN_PROC_HEADERS_LIST
+      "${DAWN_INCLUDE_DIR}/dawn/dawn_thread_dispatch_proc.h"
+      "${DAWN_INCLUDE_DIR}/dawn/dawn_proc.h"
+    )
+  endif()
   dawn_add_library(
     dawn_proc
     UTILITY_TARGET dawn_internal_config
     HEADERS
-      "${DAWN_INCLUDE_DIR}/dawn/dawn_thread_dispatch_proc.h"
-      "${DAWN_INCLUDE_DIR}/dawn/dawn_proc.h"
+      "${DAWN_PROC_HEADERS_LIST}"
     SOURCES
       "${DAWNPROC_GEN_SOURCES}"
     DEPENDS
@@ -141,6 +172,7 @@ if (NOT EMSCRIPTEN)
     PRIVATE_DEPENDS
       dawn::dawn_common
   )
+  unset(DAWN_PROC_HEADERS_LIST)
   target_compile_definitions(dawn_proc PRIVATE "WGPU_IMPLEMENTATION")
 endif()
 
diff --git a/src/dawn/glfw/CMakeLists.txt b/src/dawn/glfw/CMakeLists.txt
index f3bca8c5a1..e9a3204d9c 100644
--- a/src/dawn/glfw/CMakeLists.txt
+++ b/src/dawn/glfw/CMakeLists.txt
@@ -26,7 +26,7 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 set(headers
-    "${DAWN_INCLUDE_DIR}/webgpu/webgpu_glfw.h"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu_glfw.h>"
 )
 set(sources)
 set(conditional_public_depends)
diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index 4fa7334557..f4727529e9 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -39,6 +39,12 @@ set(headers
     "${DAWN_INCLUDE_DIR}/dawn/native/DawnNative.h"
     "${DAWN_INCLUDE_DIR}/dawn/native/dawn_native_export.h"
 )
+if(NOT DAWN_BUILD_MONOLITHIC_LIBRARY)
+    set(headers
+        "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/DawnNative.h>"
+        "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/dawn_native_export.h>"
+    )
+endif()
 
 set(private_headers
     "${DAWN_NATIVE_UTILS_GEN_HEADERS}"
@@ -324,9 +330,15 @@ if (WINDOWS_STORE)
 endif()
 
 if (DAWN_ENABLE_D3D11 OR DAWN_ENABLE_D3D12)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/D3DBackend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/D3DBackend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/D3DBackend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "d3d/BackendD3D.h"
         "d3d/BlobD3D.h"
@@ -364,9 +376,15 @@ if (DAWN_ENABLE_D3D11 OR DAWN_ENABLE_D3D12)
 endif()
 
 if (DAWN_ENABLE_D3D11)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/D3D11Backend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/D3D11Backend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/D3D11Backend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "d3d11/BackendD3D11.h"
         "d3d11/BindGroupD3D11.h"
@@ -421,9 +439,15 @@ if (DAWN_ENABLE_D3D11)
 endif()
 
 if (DAWN_ENABLE_D3D12)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/D3D12Backend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/D3D12Backend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/D3D12Backend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "d3d12/BackendD3D12.h"
         "d3d12/BindGroupD3D12.h"
@@ -509,9 +533,15 @@ if (DAWN_ENABLE_D3D12)
 endif()
 
 if (DAWN_ENABLE_METAL)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/MetalBackend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/MetalBackend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/MetalBackend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "metal/BackendMTL.h"
         "metal/BindGroupLayoutMTL.h"
@@ -575,9 +605,15 @@ if (DAWN_ENABLE_METAL)
 endif()
 
 if (DAWN_ENABLE_NULL)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/NullBackend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/NullBackend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/NullBackend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "null/DeviceNull.h"
     )
@@ -587,10 +623,17 @@ if (DAWN_ENABLE_NULL)
 endif()
 
 if (DAWN_ENABLE_WEBGPU_ON_WEBGPU)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/WebGPUBackend.h"
-        "${DAWN_SRC_DIR}/dawn/serialization/Schema.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/WebGPUBackend.h"
+            "${DAWN_SRC_DIR}/dawn/serialization/Schema.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/WebGPUBackend.h>"
+            "$<BUILD_INTERFACE:${DAWN_SRC_DIR}/dawn/serialization/Schema.h>"
+        )
+    endif()
     list(APPEND private_headers
         "webgpu/BackendWGPU.h"
         "webgpu/BindGroupWGPU.h"
@@ -664,17 +707,24 @@ if (DAWN_ENABLE_OPENGL)
     DawnGenerator(
         SCRIPT "${Dawn_SOURCE_DIR}/generator/opengl_loader_generator.py"
         PRINT_NAME "OpenGL function loader"
-        EXTRA_PARAMETERS "--gl-xml"
-             "${DAWN_OPENGL_REGISTRY_DIR}/xml/gl.xml"
-             "--supported-extensions"
-             "${Dawn_SOURCE_DIR}/src/dawn/native/opengl/supported_extensions.json"
+        EXTRA_PARAMETERS
+            "--gl-xml"
+                "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/share/opengl/gl.xml"
+            "--supported-extensions"
+                "${Dawn_SOURCE_DIR}/src/dawn/native/opengl/supported_extensions.json"
         OUTPUT_HEADERS DAWN_NATIVE_OPENGL_AUTOGEN_HEADERS
         OUTPUT_SOURCES DAWN_NATIVE_OPENGL_AUTOGEN_SOURCES
     )
 
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/OpenGLBackend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/OpenGLBackend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/OpenGLBackend.h>"
+        )
+    endif()
     list(APPEND private_headers
         ${DAWN_NATIVE_OPENGL_AUTOGEN_HEADERS}
         "opengl/BackendGL.h"
@@ -745,13 +795,20 @@ if (DAWN_ENABLE_OPENGL)
         "opengl/UtilsGL.cpp"
     )
 
-    list(APPEND conditional_private_depends dawn_khronos_platform)
+    # probably redundant
+    list(APPEND conditional_private_depends "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include") # dawn_khronos_platform
 endif()
 
 if (DAWN_ENABLE_VULKAN)
-    list(APPEND headers
-        "${DAWN_INCLUDE_DIR}/dawn/native/VulkanBackend.h"
-    )
+    if(DAWN_BUILD_MONOLITHIC_LIBRARY)
+        list(APPEND headers
+            "${DAWN_INCLUDE_DIR}/dawn/native/VulkanBackend.h"
+        )
+    else()
+        list(APPEND headers
+            "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/native/VulkanBackend.h>"
+        )
+    endif()
     list(APPEND private_headers
         "vulkan/BackendVk.h"
         "vulkan/BindGroupLayoutVk.h"
@@ -846,6 +903,8 @@ if (DAWN_ENABLE_VULKAN)
     list(APPEND conditional_public_depends
         Vulkan::Headers
         Vulkan::UtilityHeaders
+        #Vulkan::CompilerConfiguration
+        #Vulkan::LayerSettings
     )
 
     if (ANDROID)
@@ -871,12 +930,6 @@ if (DAWN_ENABLE_VULKAN)
     endif()
 endif()
 
-if (DAWN_ENABLE_D3D12 OR DAWN_ENABLE_VULKAN)
-    list(APPEND conditional_private_depends RenderDocApi)
-    list(APPEND private_headers "utils/RenderDoc.h")
-    list(APPEND sources "utils/RenderDoc.cpp")
-endif()
-
 set(dawn_component_srcs)
 list(APPEND dawn_component_srcs "DawnNative.cpp")
 if (DAWN_ENABLE_D3D11 OR DAWN_ENABLE_D3D12)
diff --git a/src/dawn/node/CMakeLists.txt b/src/dawn/node/CMakeLists.txt
index f28474bfa3..e6b6809056 100644
--- a/src/dawn/node/CMakeLists.txt
+++ b/src/dawn/node/CMakeLists.txt
@@ -89,7 +89,7 @@ endfunction()
 add_subdirectory(binding)
 add_subdirectory(interop)
 
-add_library(dawn_node SHARED
+add_library(dawn_node
     "ManualConstructors.cpp"
     "ManualConstructors.h"
     "Module.cpp"
diff --git a/src/dawn/partition_alloc/CMakeLists.txt b/src/dawn/partition_alloc/CMakeLists.txt
index bb0561394f..54f66cf98c 100644
--- a/src/dawn/partition_alloc/CMakeLists.txt
+++ b/src/dawn/partition_alloc/CMakeLists.txt
@@ -31,9 +31,9 @@ dawn_add_library(
   ENABLE_EMSCRIPTEN
   UTILITY_TARGET dawn_internal_config
   HEADERS
-    "partition_alloc/pointers/raw_ptr_exclusion.h"
-    "partition_alloc/pointers/raw_ptr.h"
-    "partition_alloc/pointers/raw_ref.h"
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/partition_alloc/pointers/raw_ptr_exclusion.h>"
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/partition_alloc/pointers/raw_ptr.h>"
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/partition_alloc/pointers/raw_ref.h>"
 )
 
-target_include_directories(partition_alloc INTERFACE ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(partition_alloc INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
diff --git a/src/dawn/platform/CMakeLists.txt b/src/dawn/platform/CMakeLists.txt
index 34648c9715..6103a922c2 100644
--- a/src/dawn/platform/CMakeLists.txt
+++ b/src/dawn/platform/CMakeLists.txt
@@ -29,8 +29,8 @@ dawn_add_library(
   dawn_platform
   UTILITY_TARGET dawn_internal_config
   HEADERS
-    "${DAWN_INCLUDE_DIR}/dawn/platform/DawnPlatform.h"
-    "${DAWN_INCLUDE_DIR}/dawn/platform/dawn_platform_export.h"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/platform/DawnPlatform.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/platform/dawn_platform_export.h>"
   PRIVATE_HEADERS
     "WorkerThread.h"
     "metrics/HistogramMacros.h"
diff --git a/src/dawn/utils/CMakeLists.txt b/src/dawn/utils/CMakeLists.txt
index 105e9a5ebd..0521aba0b5 100644
--- a/src/dawn/utils/CMakeLists.txt
+++ b/src/dawn/utils/CMakeLists.txt
@@ -59,6 +59,7 @@ dawn_add_library(
     "TextureUtils.cpp"
     "WGPUHelpers.cpp"
   DEPENDS
+    SPIRV-Tools
     webgpu_cpp
     dawn_internal_config
     dawn::dawn_common
diff --git a/src/dawn/wire/CMakeLists.txt b/src/dawn/wire/CMakeLists.txt
index bdada2bbc3..348cdd1114 100644
--- a/src/dawn/wire/CMakeLists.txt
+++ b/src/dawn/wire/CMakeLists.txt
@@ -33,10 +33,10 @@ DawnJSONGenerator(
 )
 
 set(headers
-    "${DAWN_INCLUDE_DIR}/dawn/wire/Wire.h"
-    "${DAWN_INCLUDE_DIR}/dawn/wire/WireClient.h"
-    "${DAWN_INCLUDE_DIR}/dawn/wire/WireServer.h"
-    "${DAWN_INCLUDE_DIR}/dawn/wire/dawn_wire_export.h"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/wire/Wire.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/wire/WireClient.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/wire/WireServer.h>"
+    "$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/dawn/wire/dawn_wire_export.h>"
 )
 
 set(private_headers
diff --git a/src/emdawnwebgpu/CMakeLists.txt b/src/emdawnwebgpu/CMakeLists.txt
index ad292d98bd..bb22b08f13 100644
--- a/src/emdawnwebgpu/CMakeLists.txt
+++ b/src/emdawnwebgpu/CMakeLists.txt
@@ -104,7 +104,7 @@ if (EMSCRIPTEN)
         # Note this requires Emscripten 4.0.3+ (prior to that it was located
         # at tools/maint/gen_struct_info.py).
         set(ARGS
-            ${Python3_EXECUTABLE}
+            "${Python3_EXECUTABLE}"
             "${EMSCRIPTEN_ROOT_PATH}/tools/gen_struct_info.py"
             -q
             "${EM_BUILD_GEN_DIR}/struct_info_webgpu.json"
@@ -168,7 +168,7 @@ if (EMSCRIPTEN)
             "${EM_SRC_DIR}/snippets/library_webgpu_struct_info_part3.txt"
         )
         set(ARGS
-            ${Python3_EXECUTABLE}
+            "${Python3_EXECUTABLE}"
             "${EM_SRC_DIR}/concat.py"
             "${OUTPUT}"
             ${SRCS}
@@ -192,7 +192,7 @@ if (EMSCRIPTEN)
     # Include dir, and dependency on the generated files in the include dir.
     add_library(emdawnwebgpu_c_include INTERFACE)
     target_include_directories(emdawnwebgpu_c_include BEFORE INTERFACE
-        "${EM_BUILD_GEN_DIR}/include"
+        "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include>"
     )
     add_dependencies(emdawnwebgpu_c_include emdawnwebgpu_headers_gen)
 
@@ -200,7 +200,7 @@ if (EMSCRIPTEN)
         emdawnwebgpu_c
         ENABLE_EMSCRIPTEN
         HEADERS
-            "${EM_BUILD_GEN_DIR}/include/webgpu/webgpu.h"
+            "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include/webgpu/webgpu.h>"
         SOURCES
             "${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/webgpu.cpp"
         DEPENDS
@@ -208,11 +208,15 @@ if (EMSCRIPTEN)
     )
     target_link_options(emdawnwebgpu_c INTERFACE
         # IMPORTANT: If changing these dependencies, update EMDAWNWEBGPU_C_ALL_FILE_DEPS too.
-        "--js-library=${EM_BUILD_GEN_DIR}/library_webgpu_enum_tables.js"
-        "--js-library=${EM_BUILD_GEN_DIR}/library_webgpu_generated_sig_info.js"
-        "--js-library=${EM_BUILD_GEN_DIR}/library_webgpu_generated_struct_info.js"
-        "--js-library=${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/library_webgpu.js"
-        "--closure-args=--externs=${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/webgpu-externs.js"
+        #
+        # fixing absolute paths in target_link_options() by bundling those JS files into the package
+        # and replacing paths, which should work since it is INTERFACE scope here, so they are not needed
+        # for the build
+        [=[--js-library=${_IMPORT_PREFIX}/share/Dawn/js/library_webgpu_enum_tables.js]=]
+        [=[--js-library=${_IMPORT_PREFIX}/share/Dawn/js/library_webgpu_generated_sig_info.js]=]
+        [=[--js-library=${_IMPORT_PREFIX}/share/Dawn/js/library_webgpu_generated_struct_info.js]=]
+        [=[--js-library=${_IMPORT_PREFIX}/share/Dawn/js/library_webgpu.js]=]
+        [=[--closure-args=--externs=${_IMPORT_PREFIX}/share/Dawn/js/webgpu-externs.js]=]
     )
 
     # Dependencies from emdawnwebgpu_c to the files referenced in the linker flags.
@@ -220,13 +224,13 @@ if (EMSCRIPTEN)
     #   target-level dependencies (defining *how* to rebuild).
     set(EMDAWNWEBGPU_C_GENERATED_FILE_DEPS
         "${EMDAWNWEBGPU_JS_GEN_SOURCES}"
-        "${EMDAWNWEBGPU_STRUCT_INFO_JS}"
+        "$<BUILD_INTERFACE:${EMDAWNWEBGPU_STRUCT_INFO_JS}>"
     )
     # - There are also source files, which are used only as file-level dependencies.
     set(EMDAWNWEBGPU_C_ALL_FILE_DEPS
         "${EMDAWNWEBGPU_C_GENERATED_FILE_DEPS}"
-        "${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/library_webgpu.js"
-        "${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/webgpu-externs.js"
+        "$<BUILD_INTERFACE:${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/library_webgpu.js>"
+        "$<BUILD_INTERFACE:${DAWN_EMDAWNWEBGPU_DIR}/pkg/webgpu/src/webgpu-externs.js>"
     )
     add_custom_target(emdawnwebgpu_config_generated_deps
         DEPENDS "${EMDAWNWEBGPU_C_GENERATED_FILE_DEPS}"
@@ -242,10 +246,11 @@ if (EMSCRIPTEN)
         ENABLE_EMSCRIPTEN
         HEADER_ONLY
         HEADERS
-            "${EM_BUILD_GEN_DIR}/include/dawn/webgpu_cpp_print.h"
-            "${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_cpp.h"
-            "${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_cpp_chained_struct.h"
-            "${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_enum_class_bitmasks.h"
+            "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include/dawn/webgpu_cpp_print.h>"
+            "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_cpp.h>"
+            "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_cpp_chained_struct.h>"
+            #"$<BUILD_INTERFACE:${DAWN_INCLUDE_DIR}/webgpu/webgpu_enum_class_bitmasks.h>"
+            "$<BUILD_INTERFACE:${EM_BUILD_GEN_DIR}/include/webgpu/webgpu_enum_class_bitmasks.h>"
         DEPENDS
             emdawnwebgpu_c
     )
diff --git a/src/tint/CMakeLists.txt b/src/tint/CMakeLists.txt
index e30d31c7ad..0fcb124bc4 100644
--- a/src/tint/CMakeLists.txt
+++ b/src/tint/CMakeLists.txt
@@ -59,8 +59,8 @@ endif()
 # Helper functions
 ################################################################################
 function(tint_core_compile_options TARGET)
-  target_include_directories(${TARGET} PUBLIC "${TINT_ROOT_SOURCE_DIR}")
-  target_include_directories(${TARGET} PUBLIC "${TINT_ROOT_SOURCE_DIR}/include")
+  target_include_directories(${TARGET} PUBLIC "$<BUILD_INTERFACE:${TINT_ROOT_SOURCE_DIR}>")
+  #target_include_directories(${TARGET} PUBLIC "${TINT_ROOT_SOURCE_DIR}/include")
   target_compile_definitions(${TARGET} PUBLIC -DTINT_BUILD_GLSL_VALIDATOR=$<BOOL:${TINT_BUILD_GLSL_VALIDATOR}>)
   target_compile_definitions(${TARGET} PUBLIC -DTINT_BUILD_GLSL_WRITER=$<BOOL:${TINT_BUILD_GLSL_WRITER}>)
   target_compile_definitions(${TARGET} PUBLIC -DTINT_BUILD_HLSL_WRITER=$<BOOL:${TINT_BUILD_HLSL_WRITER}>)
@@ -220,32 +220,48 @@ function(tint_default_compile_options TARGET)
 endfunction()
 
 function(tint_spvheaders_compile_options TARGET)
-  target_link_libraries(${TARGET} PRIVATE SPIRV-Headers)
-  target_include_directories(${TARGET} PRIVATE "${TINT_SPIRV_HEADERS_DIR}/include")
+  target_link_libraries(${TARGET} PRIVATE SPIRV-Headers::SPIRV-Headers)
 endfunction()
 
 function(tint_spvtools_compile_options TARGET)
   target_link_libraries(${TARGET} PRIVATE SPIRV-Tools)
-  target_include_directories(${TARGET} PRIVATE "${TINT_SPIRV_TOOLS_DIR}/include")
 endfunction()
 
 function(tint_lib_compile_options TARGET)
-  if (TINT_ENABLE_INSTALL)
-    install(TARGETS ${TARGET}
-      LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    )
-  endif()
+  # this does nothing good (does not export its own targets and would miss other targets),
+  # we have our own installation in the main CMakeLists.txt
+  #if (TINT_ENABLE_INSTALL)
+  #  install(TARGETS ${TARGET}
+  #    # a good place to do this, but it won't work, because it will require
+  #    # certain other targets to be exported (such as dawn_shared_utils),
+  #    # which has to happen elsewhere, so unfortunately we keep maintaining
+  #    # that list of export/install targets in the main CMakeLists.txt
+  #    #EXPORT "tintTargets"
+  #    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  #    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  #  )
+  #  # a good place to do this, but it won't work, because it will require
+  #  # certain other targets to be exported (such as dawn_shared_utils),
+  #  # which has to happen elsewhere, so unfortunately we keep maintaining
+  #  # that list of export/install targets in the main CMakeLists.txt
+  #  #install(EXPORT "tintTargets"
+  #  #    FILE "tintTargets.cmake"
+  #  #    NAMESPACE "dawn::"
+  #  #    DESTINATION "share/${PROJECT_NAME}"
+  #  #)
+  #endif()
   tint_default_compile_options(${TARGET})
 endfunction()
 
 function(tint_proto_compile_options TARGET)
-  if (TINT_ENABLE_INSTALL)
-    install(TARGETS ${TARGET}
-      LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    )
-  endif()
+  # this does nothing good (does not export its own targets and would miss other targets),
+  # we have our own installation in the main CMakeLists.txt
+  #if (TINT_ENABLE_INSTALL)
+  #  install(TARGETS ${TARGET}
+  #    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  #    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  #  )
+  #endif()
   tint_core_compile_options(${TARGET})
 endfunction()
 
@@ -410,7 +426,7 @@ function(tint_add_target TARGET KIND)
   endif()
 
   if(${KIND} STREQUAL lib)
-    add_library(${TARGET} STATIC EXCLUDE_FROM_ALL)
+    add_library(${TARGET} STATIC)
     tint_lib_compile_options(${TARGET})
   elseif(${KIND} STREQUAL proto)
     add_library(${TARGET} STATIC EXCLUDE_FROM_ALL)
@@ -531,7 +547,8 @@ function(tint_target_add_external_dependencies TARGET KIND)
     elseif(${DEPENDENCY} STREQUAL "dl")
       target_link_libraries(${TARGET} PRIVATE ${CMAKE_DL_LIBS})
     elseif(${DEPENDENCY} STREQUAL "dxc-include")
-      target_include_directories(${TARGET} PRIVATE "${DAWN_THIRD_PARTY_DIR}/dxc/include")
+      find_package(D3D12 REQUIRED)
+      target_include_directories(${TARGET} PRIVATE ${D3D12_INCLUDE_DIR}) # ${D3D12_INCLUDE_DIRS} for the full list
     elseif(${DEPENDENCY} STREQUAL "dxcompiler-for-fuzzer")
       if(TINT_BUILD_FUZZERS AND DAWN_USE_BUILT_DXC) # TODO: and target arch is not x86
         target_link_libraries(${TARGET} PRIVATE dxcompiler)
@@ -584,12 +601,6 @@ function(tint_target_add_external_dependencies TARGET KIND)
       target_link_libraries(${TARGET} PRIVATE
         SPIRV-Tools-opt
       )
-      target_include_directories(${TARGET} PRIVATE
-        "${TINT_SPIRV_TOOLS_DIR}"
-        "${TINT_SPIRV_TOOLS_DIR}/include"
-        "${TINT_SPIRV_TOOLS_DIR}/source"
-        "${spirv-tools_BINARY_DIR}"
-      )
     elseif(${DEPENDENCY} STREQUAL "thread")
       find_package(Threads REQUIRED)
       target_link_libraries(${TARGET} PRIVATE Threads::Threads)
@@ -691,6 +702,11 @@ else()
   include("BUILD.cmake")
 endif(TINT_BUILD_FUZZERS)
 
+target_include_directories(tint_utils_macros
+    PUBLIC
+        "$<BUILD_INTERFACE:${Dawn_SOURCE_DIR}/src>"
+        "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+)
 
 ################################################################################
 # Generate protobuf sources
@@ -748,16 +764,18 @@ add_library(libtint ALIAS tint_api)
 ################################################################################
 # Install rules
 ################################################################################
-if (TINT_ENABLE_INSTALL)
-  # We need to recurse all folders and include all headers because tint.h includes most headers from src/tint
-  file(GLOB TINT_HEADERS "${DAWN_INCLUDE_DIR}/tint/*.h")
-
-  install(FILES ${TINT_HEADERS}  DESTINATION  ${CMAKE_INSTALL_INCLUDEDIR}/tint/)
-
-  file(GLOB_RECURSE TINT_SRC_HEADERS RELATIVE ${TINT_ROOT_SOURCE_DIR} "*.h")
-
-  foreach(TINT_HEADER_FILE ${TINT_SRC_HEADERS})
-      get_filename_component(TINT_HEADER_DIR ${TINT_HEADER_FILE} DIRECTORY)
-      install(FILES ${TINT_ROOT_SOURCE_DIR}/${TINT_HEADER_FILE}  DESTINATION  ${CMAKE_INSTALL_INCLUDEDIR}/src/tint/${TINT_HEADER_DIR})
-  endforeach ()
-endif()
+# this does nothing good, we have our own installation in the main CMakeLists.txt
+#if (TINT_ENABLE_INSTALL)
+#  # We need to recurse all folders and include all headers because tint.h includes most headers from src/tint
+#  file(GLOB TINT_HEADERS "${DAWN_INCLUDE_DIR}/tint/*.h")
+#
+#  install(FILES ${TINT_HEADERS}  DESTINATION  ${CMAKE_INSTALL_INCLUDEDIR}/tint/)
+#
+#  file(GLOB_RECURSE TINT_SRC_HEADERS RELATIVE ${TINT_ROOT_SOURCE_DIR} "*.h")
+#
+#  foreach(TINT_HEADER_FILE ${TINT_SRC_HEADERS})
+#      get_filename_component(TINT_HEADER_DIR ${TINT_HEADER_FILE} DIRECTORY)
+#      # the `/src/tint/` part of the path should be removed, how was this ever going to work
+#      install(FILES ${TINT_ROOT_SOURCE_DIR}/${TINT_HEADER_FILE}  DESTINATION  ${CMAKE_INSTALL_INCLUDEDIR}/src/tint/${TINT_HEADER_DIR})
+#  endforeach ()
+#endif()
diff --git a/src/tint/api/tint_api_test.cc b/src/tint/api/tint_api_test.cc
index d7254f060f..3075f20a2a 100644
--- a/src/tint/api/tint_api_test.cc
+++ b/src/tint/api/tint_api_test.cc
@@ -30,7 +30,7 @@
 #include "src/tint/api/tint.h"
 
 #if TINT_BUILD_SPV_READER
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #endif
 
 namespace tint {
diff --git a/src/tint/cmd/common/helper.cc b/src/tint/cmd/common/helper.cc
index e577497a23..5f68a993b8 100644
--- a/src/tint/cmd/common/helper.cc
+++ b/src/tint/cmd/common/helper.cc
@@ -38,7 +38,7 @@
 #endif
 
 #if TINT_BUILD_SPV_READER
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/lang/spirv/reader/reader.h"
 #endif
 
diff --git a/src/tint/cmd/fuzz/ir/dis/main.cc b/src/tint/cmd/fuzz/ir/dis/main.cc
index b63d8c7e7c..a66956c8d5 100644
--- a/src/tint/cmd/fuzz/ir/dis/main.cc
+++ b/src/tint/cmd/fuzz/ir/dis/main.cc
@@ -30,7 +30,7 @@
 #include <memory>
 #include <string>
 
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/api/tint.h"
 #include "src/tint/cmd/common/helper.h"
 #include "src/tint/lang/core/ir/binary/decode.h"
diff --git a/src/tint/cmd/tint/main.cc b/src/tint/cmd/tint/main.cc
index 5a31670be4..751295337c 100644
--- a/src/tint/cmd/tint/main.cc
+++ b/src/tint/cmd/tint/main.cc
@@ -34,7 +34,7 @@
 #include <vector>
 
 #if TINT_BUILD_SPV_READER || TINT_BUILD_SPV_WRITER
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #endif  // TINT_BUILD_SPV_READER || TINT_BUILD_SPV_WRITER
 
 #include "src/tint/api/common/substitute_overrides_config.h"
diff --git a/src/tint/lang/hlsl/validate/validate.cc b/src/tint/lang/hlsl/validate/validate.cc
index 1a50bfacf0..a01c6511ee 100644
--- a/src/tint/lang/hlsl/validate/validate.cc
+++ b/src/tint/lang/hlsl/validate/validate.cc
@@ -51,7 +51,7 @@ TINT_BEGIN_DISABLE_ALL_WARNINGS();
 // # Use UUID emulation with clang to avoid compiling with ms-extensions
 #define __EMULATE_UUID
 #endif
-#include "dxc/dxcapi.h"
+#include <dxcapi.h>
 TINT_END_DISABLE_ALL_WARNINGS();
 
 // Disable warnings about old-style casts which result from using
diff --git a/src/tint/lang/spirv/reader/common/helper_test.h b/src/tint/lang/spirv/reader/common/helper_test.h
index 8c901971ed..a44f253b83 100644
--- a/src/tint/lang/spirv/reader/common/helper_test.h
+++ b/src/tint/lang/spirv/reader/common/helper_test.h
@@ -31,7 +31,7 @@
 #include <string>
 #include <vector>
 
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/utils/result.h"
 
 namespace tint::spirv::reader {
diff --git a/src/tint/lang/spirv/reader/parser/parser.cc b/src/tint/lang/spirv/reader/parser/parser.cc
index 6f45c85094..e4ca7a4ab1 100644
--- a/src/tint/lang/spirv/reader/parser/parser.cc
+++ b/src/tint/lang/spirv/reader/parser/parser.cc
@@ -47,9 +47,9 @@ TINT_BEGIN_DISABLE_WARNING(OLD_STYLE_CAST);
 TINT_BEGIN_DISABLE_WARNING(SIGN_CONVERSION);
 TINT_BEGIN_DISABLE_WARNING(WEAK_VTABLES);
 TINT_BEGIN_DISABLE_WARNING(UNSAFE_BUFFER_USAGE);
-#include "source/opt/build_module.h"
-#include "source/opt/resolve_binding_conflicts_pass.h"
-#include "source/opt/split_combined_image_sampler_pass.h"
+#include <spirv-tools/source/opt/build_module.h>
+#include <spirv-tools/source/opt/resolve_binding_conflicts_pass.h>
+#include <spirv-tools/source/opt/split_combined_image_sampler_pass.h>
 TINT_END_DISABLE_WARNING(UNSAFE_BUFFER_USAGE);
 TINT_END_DISABLE_WARNING(WEAK_VTABLES);
 TINT_END_DISABLE_WARNING(SIGN_CONVERSION);
diff --git a/src/tint/lang/spirv/validate/validate.cc b/src/tint/lang/spirv/validate/validate.cc
index b05dc76171..093f73c526 100644
--- a/src/tint/lang/spirv/validate/validate.cc
+++ b/src/tint/lang/spirv/validate/validate.cc
@@ -32,7 +32,7 @@
 #include <string>
 #include <utility>
 
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/utils/diagnostic/diagnostic.h"
 
 namespace tint::spirv::validate {
diff --git a/src/tint/lang/spirv/validate/validate.h b/src/tint/lang/spirv/validate/validate.h
index 10c044285b..9c87ca1ff1 100644
--- a/src/tint/lang/spirv/validate/validate.h
+++ b/src/tint/lang/spirv/validate/validate.h
@@ -30,7 +30,7 @@
 
 #include <span>
 
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/utils/result.h"
 
 namespace tint::spirv::validate {
diff --git a/src/tint/lang/spirv/writer/common/helper_test.h b/src/tint/lang/spirv/writer/common/helper_test.h
index 18ababc93b..80c3181b6f 100644
--- a/src/tint/lang/spirv/writer/common/helper_test.h
+++ b/src/tint/lang/spirv/writer/common/helper_test.h
@@ -34,7 +34,7 @@
 
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/lang/core/ir/builder.h"
 #include "src/tint/lang/core/ir/disassembler.h"
 #include "src/tint/lang/core/ir/validator.h"
diff --git a/src/tint/lang/spirv/writer/common/spv_dump_test.cc b/src/tint/lang/spirv/writer/common/spv_dump_test.cc
index b69923ab69..ae1053f6fb 100644
--- a/src/tint/lang/spirv/writer/common/spv_dump_test.cc
+++ b/src/tint/lang/spirv/writer/common/spv_dump_test.cc
@@ -27,7 +27,7 @@
 
 #include "src/tint/lang/spirv/writer/common/spv_dump_test.h"
 
-#include "spirv-tools/libspirv.hpp"
+#include <spirv-tools/libspirv.hpp>
 #include "src/tint/lang/spirv/writer/common/binary_writer.h"
 
 namespace tint::spirv::writer {
diff --git a/src/tint/utils/macros/compiler.h b/src/tint/utils/macros/compiler.h
index ce3b439ffa..fcb291d045 100644
--- a/src/tint/utils/macros/compiler.h
+++ b/src/tint/utils/macros/compiler.h
@@ -26,7 +26,7 @@
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include "src/tint/utils/macros/concat.h"
-#include "src/utils/compiler.h"
+#include "utils/compiler.h"
 
 #ifndef SRC_TINT_UTILS_MACROS_COMPILER_H_
 #define SRC_TINT_UTILS_MACROS_COMPILER_H_
diff --git a/src/utils/CMakeLists.txt b/src/utils/CMakeLists.txt
index 66d4bf6604..a0c8ea8703 100644
--- a/src/utils/CMakeLists.txt
+++ b/src/utils/CMakeLists.txt
@@ -26,8 +26,8 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 set(private_headers
-    "compiler.h"
-    "numeric.h"
+    "$<INSTALL_INTERFACE:compiler.h>"
+    "$<INSTALL_INTERFACE:numeric.h>"
 )
 
 set(sources
diff --git a/tools/android/webgpu/src/main/cpp/CMakeLists.txt b/tools/android/webgpu/src/main/cpp/CMakeLists.txt
index 6d5296fbb0..61fea32f82 100644
--- a/tools/android/webgpu/src/main/cpp/CMakeLists.txt
+++ b/tools/android/webgpu/src/main/cpp/CMakeLists.txt
@@ -50,7 +50,7 @@ DawnJSONGenerator(
     OUTPUT_HEADERS WEBGPU_JNI_HEADERS
 )
 
-add_library(webgpu_c_bundled SHARED)
+add_library(webgpu_c_bundled)
 target_sources(webgpu_c_bundled PRIVATE
     ${WEBGPU_JNI_SOURCES}
     ${WEBGPU_JNI_HEADERS}
