diff --git a/src/dawn/native/Device.h b/src/dawn/native/Device.h
index 0f6277d612..400bc15047 100644
--- a/src/dawn/native/Device.h
+++ b/src/dawn/native/Device.h
@@ -53,6 +53,7 @@
 
 #include "dawn/native/DawnNative.h"
 #include "dawn/native/dawn_platform.h"
+#include "tint/lang/wgsl/common/allowed_features.h"
 
 namespace dawn::platform {
 class WorkerTaskPool;
diff --git a/src/dawn/native/ObjectBase.h b/src/dawn/native/ObjectBase.h
index c98861e728..61059494d5 100644
--- a/src/dawn/native/ObjectBase.h
+++ b/src/dawn/native/ObjectBase.h
@@ -36,9 +36,7 @@
 #include "dawn/common/RefCounted.h"
 #include "dawn/native/Forward.h"
 
-namespace absl {
-class FormatSink;
-}
+#include <absl/strings/str_format.h>
 
 namespace dawn::native {
 
diff --git a/src/dawn/native/metal/ShaderModuleMTL.mm b/src/dawn/native/metal/ShaderModuleMTL.mm
index 80c561c86f..673a0b2a87 100644
--- a/src/dawn/native/metal/ShaderModuleMTL.mm
+++ b/src/dawn/native/metal/ShaderModuleMTL.mm
@@ -43,6 +43,7 @@
 #include "dawn/platform/tracing/TraceEvent.h"
 
 #include <tint/tint.h>
+#include "tint/lang/msl/writer/writer.h"
 
 #include <sstream>
 
diff --git a/src/dawn/native/vulkan/ShaderModuleVk.cpp b/src/dawn/native/vulkan/ShaderModuleVk.cpp
index f27f4d905c..57d2166b7d 100644
--- a/src/dawn/native/vulkan/ShaderModuleVk.cpp
+++ b/src/dawn/native/vulkan/ShaderModuleVk.cpp
@@ -47,6 +47,7 @@
 #include "dawn/platform/metrics/HistogramMacros.h"
 #include "dawn/platform/tracing/TraceEvent.h"
 #include "tint/tint.h"
+#include "tint/lang/wgsl/reader/reader.h"
 
 #ifdef DAWN_ENABLE_SPIRV_VALIDATION
 #include "dawn/native/SpirvValidation.h"
diff --git a/src/tint/lang/spirv/reader/reader.h b/src/tint/lang/spirv/reader/reader.h
index 9bcd99e550..02d23442a5 100644
--- a/src/tint/lang/spirv/reader/reader.h
+++ b/src/tint/lang/spirv/reader/reader.h
@@ -32,6 +32,7 @@
 
 #include "src/tint/lang/spirv/reader/common/options.h"
 #include "src/tint/lang/wgsl/program/program.h"
+#include "src/tint/lang/core/ir/module.h"
 
 // Forward declarations
 namespace tint::core::ir {
