diff --git a/CMakeLists.txt b/CMakeLists.txt
index df42815b..a04da006 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -302,7 +302,9 @@ include(GNUInstallDirs)
 
 set(PROJECT_NAME "${PROJECT_NAME_ORIGINAL}")
 
-set(PROJ_DATA_PATH "${CMAKE_INSTALL_FULL_DATADIR}/proj")
+if(NOT DEFINED PROJ_DATA_PATH)
+    set(PROJ_DATA_PATH "${CMAKE_INSTALL_FULL_DATADIR}/proj")
+endif()
 
 ################################################################################
 # Tests
@@ -323,7 +325,7 @@ include_directories(${PROJ_SOURCE_DIR}/src)
 add_subdirectory(data)
 add_subdirectory(include)
 add_subdirectory(src)
-add_subdirectory(man)
+#add_subdirectory(man)
 add_subdirectory(cmake)
 if(BUILD_TESTING)
   add_subdirectory(test)
@@ -334,18 +336,18 @@ if(BUILD_EXAMPLES)
   add_subdirectory(examples)
 endif()
 
-set(docfiles COPYING NEWS AUTHORS)
-install(FILES ${docfiles}
-        DESTINATION ${CMAKE_INSTALL_DOCDIR})
+#set(docfiles COPYING NEWS AUTHORS)
+#install(FILES ${docfiles}
+#        DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 ################################################################################
 # pkg-config support
 ################################################################################
 configure_proj_pc()
 
-install(FILES
-  ${CMAKE_CURRENT_BINARY_DIR}/proj.pc
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+# install(FILES
+#   ${CMAKE_CURRENT_BINARY_DIR}/proj.pc
+#   DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
 ################################################################################
 # "make dist" workalike
diff --git a/cmake/project-config.cmake.in b/cmake/project-config.cmake.in
index a9997c43..452441de 100644
--- a/cmake/project-config.cmake.in
+++ b/cmake/project-config.cmake.in
@@ -63,15 +63,15 @@ set_variable_from_rel_or_absolute_path("@PROJECT_VARIANT_NAME@_BINARY_DIRS" "${_
 set (@PROJECT_VARIANT_NAME@_LIBRARIES @PROJECT_VARIANT_NAME@::proj)
 # Read in the exported definition of the library
 include ("${_DIR}/@PROJECT_NAME_LOWER@-targets.cmake")
-if (@INSTALL_LEGACY_CMAKE_FILES@)
-  include ("${_DIR}/@PROJECT_LEGACY_LOWER@-targets.cmake")
-endif()
+# if (@INSTALL_LEGACY_CMAKE_FILES@)
+#   include ("${_DIR}/@PROJECT_LEGACY_LOWER@-targets.cmake")
+# endif()
 
 unset (_ROOT)
 unset (_DIR)
 
-if ("@PROJECT_VARIANT_NAME@" STREQUAL "PROJ4")
-  # For backward compatibility with old releases of libgeotiff
-  set (@PROJECT_VARIANT_NAME@_INCLUDE_DIR
-    ${@PROJECT_VARIANT_NAME@_INCLUDE_DIRS})
-endif ()
+# if ("@PROJECT_VARIANT_NAME@" STREQUAL "PROJ4")
+#   # For backward compatibility with old releases of libgeotiff
+#   set (@PROJECT_VARIANT_NAME@_INCLUDE_DIR
+#     ${@PROJECT_VARIANT_NAME@_INCLUDE_DIRS})
+# endif ()
diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index c49310f1..4d676032 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -498,7 +498,7 @@ install(TARGETS proj
 
 if(NOT BUILD_FRAMEWORKS_AND_BUNDLE)
   install(FILES ${ALL_LIBPROJ_HEADERS}
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/proj)
 endif()
 
 ##############################################
